\documentclass[]{article}

\usepackage{natbib}
\bibliographystyle{amnatnat}

\usepackage{amsmath,amssymb,bm}

\usepackage[colorlinks=true, citecolor=blue, urlcolor=blue, linkcolor=blue]{hyperref}

\usepackage{graphicx}
\usepackage{grffile}
\graphicspath{{../figures/}}
\usepackage{epstopdf}
\epstopdfsetup{outdir=./}

\usepackage{tikz}
\usetikzlibrary{positioning}
\usetikzlibrary{calc}
\usepackage{pgfmath,pgffor}
\usepackage{ifthen}

\usepackage{color}
\definecolor{darkblue}{rgb}{0.2472, 0.24, 0.6}
\definecolor{darkred}{rgb}{0.6, 0.24, 0.442893}
\definecolor{darkyellow}{rgb}{0.6, 0.547014, 0.24}

\usepackage{fullpage}
\usepackage{lineno}
\usepackage{setspace}

\usepackage{authblk}
\makeatletter
\let\@fnsymbol\@arabic
\makeatother

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\title{Genetic signatures of evolutionary rescue by a selective sweep}
\author[*]{Matthew M.\ Osmond\thanks{mmosmond@ucdavis.edu}}
\author[*]{Graham Coop}
\affil[*]{Center for Population Biology and Department of Evolution and Ecology, University of California, Davis, California 95616}
\date{}
\renewcommand\Affilfont{\itshape\small}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{document}
%\doublespacing

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\maketitle

%\linenumbers
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Abstract}
\label{sec:abstract}

One of the most useful models in population genetics is that of a selective sweep and the consequent hitch-hiking of linked neutral alleles.
While variations on this model typically assume constant population size, many instances of strong selection and rapid adaptation in nature may co-occur with complex demography.
Here we extend the hitch-hiking model to evolutionary rescue, where adaptation and demography not only co-occur but are intimately entwined. 
Our results show how this feedback between demography and evolution determines -- and restricts -- the genetic signatures of evolutionary rescue, and how these differ from the signatures of sweeps in populations of constant size.
In particular, we find rescue to harden sweeps from standing variance or new mutation (but not from migration), reduce nucleotide diversity both at the selected site and genome-wide, and increase the range of observed Tajima's $D$ values.
For a given initial rate of population decline, the feedback between demography and evolution makes all of these differences more dramatic under weaker selection, where bottlenecks are prolonged. 
Nevertheless, it is likely difficult to infer the co-incident timing of the sweep and bottleneck from these simple signatures, never-mind a feedback between them. 
Temporal samples spanning contemporary rescue events may offer one way forward.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Introduction}
\label{sec:intro}

The simple models used to predict the genetic signatures of selective sweeps have been incredibly helpful in understanding and identifying population genetic signals of adaptation (e.g., \citealp{smith1974hitch,kaplan1989hitchhiking}; reviewed in \citealp{stephan2019selective}). 
These models are usually based on constant-sized, Wright-Fisher populations.
Meanwhile, many instances of adaptation -- and thus selective sweeps -- in nature will co-occur with complex demography.
In fact, many of the most well-known examples of selective sweeps have arisen following a rather extreme and sudden change in the environment (e.g., after the application of insecticides, \citealp{sedghifar2016genomic}, or antimalarial drugs, \citealp{nair2003selective}), which could have simultaneously imposed sharp demographic declines.
Attempts to capture such complex demographic scenarios typically impose qualitatively appropriate changes in population size \citep[e.g.,][]{hermisson2005soft}.
Indeed, a number of studies have explored the genetic signatures of selective sweeps during demographic bottlenecks \citep[e.g.,][]{innan2004pattern,teshima2006reliable,wilson2014soft}.
However, these demographies are nearly always chosen in the absence of an explicit population model and independently of evolution.

Here we model selective sweeps in a scenario where demography and adaptive evolution are not independent.
In particular we model an instance of evolutionary rescue (\citealp{gomulkiewicz1995does}; reviewed in \citealp{bell2017evolutionary}), where a sudden environmental change causes population decline that is reverted by a selective sweep.
Under this framework, rescue is a simultaneous demographic bottleneck and selective sweep, where each affects the other.
First, because the mean absolute fitness of the population changes with the beneficial allele's frequency, the depth and duration of the bottleneck depends on the dynamics of the selective sweep, i.e., evolution affects demography.
Second, the probability that the beneficial allele establishes depends on its growth rate and thus the rate of population decline, i.e., demography affects evolution.
Together, this feedback between demography and evolution restricts the range of dynamics that are possible, and therefore also restricts the range of genetic signatures we should expect to observe.
Our goal here is to describe the range of genetic signatures that (this model of) evolutionary rescue allows, to help elucidate how rescue may obscure inferences of past selection and demography and to identify patterns that could be used to infer rescue in nature.

Most theory on evolutionary rescue to date \citep[reviewed in][]{alexander2014evolutionary} has focused on the probability of rescue.
Recently, however, some attention has been given to the dynamics of population size \citep{orr2014population}, the probability of soft sweeps \citep{wilson2017soft}, and the genetic basis of adaptation \citep{osmond2020genetic} \textit{given} rescue in haploid or asexual populations.
Here we extend this line of thinking to three modes of rescue in diploid, sexual populations, and use coalescent theory and simulations of whole chromosomes to examine the genetic signatures at linked, neutral loci.
Our focus is on three common genetic signatures: the number of unique lineages of the beneficial allele that establish (i.e., the softness of the sweep), the pattern of nucleotide diversity around the selected site (i.e., the dip in diversity), and the pattern of Tajima's $D$ around the selected site (i.e., skews in the site-frequency spectrum).
We explore three modes of rescue, where the beneficial allele arises from either standing genetic variance, recurrent \textit{de novo} mutation, or migration, and compare to selective sweeps arising in populations of constant size.
Qualitatively, we find that for a given selection coefficient rescue causes faster, harder sweeps that produce wider, deeper dips in diversity and more extreme values of Tajima's $D$.
Due to the feedback between demography and evolution, the effect of rescue on the signatures of selective sweeps becomes more pronounced as the selection coefficient, and thus the probability of rescue, gets smaller.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Methods}
\label{sec:methods}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Population dynamics}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Deterministic trajectories}
\label{sec:rescue_forward}

Consider a population of size $N(t)$ with a beneficial allele, $A$, at frequency $p(t)$ and an ancestral allele, $a$, at frequency $q(t)=1-p(t)$.
Assume non-overlapping generations and let $W_{AA}$, $W_{Aa}$, and $W_{aa}$ be the absolute fitness (expected number of offspring) of each genotype.
Then with random mating the expected change in allele frequency \citep[equation 5.2.13 in][]{crow1970introduction} and population size in one generation is
\begin{equation}\label{eq:deltapdeltaN}
\begin{aligned}
\Delta p(t) &\equiv p(t+1) - p(t) = p(t) q(t) \frac{  (W_{AA} - W_{Aa} )p(t) +  (W_{Aa} - W_{aa})q(t)}{\overline{W}(t)}  \\
\Delta N(t) &\equiv N(t+1) - N(t) = (\overline{W}(t) - 1) N(t),
\end{aligned}
\end{equation}
with $\overline{W}(t)$ the population mean fitness.

Here we are interested in the scenario where a population composed of primarily $aa$ genotypes is declining at some rate $d$, i.e., $W_{aa} = 1-d$.
The beneficial allele, $A$, is then assumed to act multiplicatively with the fitness of the ancestral background, such that $W_{Aa} = (1-d)(1+ h s)$ and $W_{AA} = (1-d)(1+s)$.
Throughout the text we assume additivity at the selected locus, $h=1/2$, such that the allele frequency \citep[c.f., equation 5.3.12 in][]{crow1970introduction} and population dynamics can be approximated by
\begin{equation}\label{eq:ptnt}
\begin{aligned}
p(t) &\approx \left( 1 + \frac{1-p(0)}{p(0)} e^{-s t/2} \right)^{-1}\\
N(t) &\approx N(0) e^{ \left[ (1+s)(1-d)-1\right]t} \left(\frac{p(0)}{p(t)} \right)^{2(1-d)}.
\end{aligned}
\end{equation}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Effective initial allele frequency}
\label{sec:initial_frequency}

Equations \ref{eq:deltapdeltaN} and \ref{eq:ptnt} consider only the deterministic dynamics. 
We are, however, primarily concerned with a stochastic event -- the establishment of a (potentially rare) beneficial allele.
In rescue we are particularly interested in only those instances where the beneficial allele establishes (otherwise the population is extinct), which creates a bias away from the deterministic trajectory.
As shown in \cite{smith1971use} \citep[see][for an application to evolutionary rescue]{orr2014population}, we can approximate this bias by assuming that a single copy of an allele that establishes with probability $P_\mathrm{est}$ instantaneously increases from a frequency of $1/(2N)$ to $1/(2NP_\mathrm{est})$.
In essence, dividing the initial allele frequency by the probability of establishment implies that an allele escaping random loss when rare will quickly increase to a larger frequency than expected before settling into its deterministic trajectory.
As we will see below, we can combine this logic with the waiting time distribution for an establishing mutation to arrive to derive an effective initial allele frequency, $p_0$.
Using the deterministic equations (Equation \ref{eq:deltapdeltaN} or \ref{eq:ptnt}) with $p(0) = p_0$ then approximates the forward-time trajectory of an allele that is destined to fixation.

We derive the probability of establishment, $P_\mathrm{est}$, assuming alleles do not interact (i.e., a branching process).
Under this assumption a single copy of an allele that is expected to leave $1+\epsilon$ copies in the next generation with a variance of $v$ has an establishment probability of \citep[][p.\ 172, see also \citealp{feller1951diffusion}, equation 5.7, for a derivation from a diffusion approximation]{allen2010introduction}
\begin{equation}\label{eq:pest}
P_\mathrm{est} = 1 - e^{-2 \epsilon/v}.
\end{equation}
For example, in a Wright-Fisher population of size $N$ a rare allele with selection advantage $s$ is expected to leave $1+s$ copies, with a variance of $1+s+O(1/N)$.
Thus $\epsilon=s$ and, in a large population with weak selection, $v \approx 1$, such that the probability of establishment is roughly $1-e^{2s}$ \citep[][p.\ 80]{fisher1999genetical} or nearly $2s$ \citep{haldane1927mathematical}. 
In our case, in a declining population ($d>0$) we have $\epsilon = W_{Aa} - 1 = (1 + s h)(1-d)-1 \approx s h - d$ and $v$ depends on the particular choice of lifecycle (e.g., see \nameref{sec:pestNe}).
When $h=1$ and $\sigma^2=1$, we recover the probability of establishment of a weakly beneficial mutation in an exponentially growing or declining haploid population with a Poisson offspring distribution, $P_\mathrm{est} \approx 2(s - d)$ \citep{otto1997probability}.
In a population at carrying capacity (e.g., with $d\geq0$) the growth rate is instead $\epsilon = W_{Aa}/W_{aa} - 1 = (1 + s h)-1 = s h$. 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Effective final allele frequency}
\label{sec:final_frequency}

Above we have stitched together the initial stochastic phase of the allele frequency and population dynamics with the deterministic phase by using $p(0)=p_0$. 
We next incorporate the final phase, fixation, which is also stochastic.
Incorporating this final phase is important here \citep[as opposed to studies that predict only the forward-time dynamics; e.g.,][]{orr2014population} because in looking at genetic signatures we are primarily concerned with the dynamics looking backwards in time from the point of fixation, and this third phase determines how fast fixation occurs.
As shown in \cite{martin2015simple}, we can treat the backward-in-time dynamics from fixation as we did the forward-in-time dynamics at establishment. 
In particular, the probability the heterozygote establishes (backward in time) in a population of mutant homozygotes is $P_\mathrm{est}$ (Equation \ref{eq:pest}) with the growth rate of a rare heterozygote in this population, $\epsilon$, replaced by $-\epsilon$ (let's call this probability $P_\mathrm{est}^-$).
For instance, if the fitness of the heterozygote is $W_{Aa} = (1-d)(1+ s h)$ and the fitness of the mutant homozygote is $W_{AA} = (1-d)(1+s)$ and the population is at carrying capacity then $-\epsilon = 1 - W_{Aa}/W_{AA}=1 - (1+ s h)/(1+ s) \approx s(1-h)$.
This does not depend on demography, $d$, in contrast to the growth rate of the heterozygote at establishment (see \nameref{sec:initial_frequency}).
Note that the variance in the number of offspring, $v$, may also be different at the time of fixation than it was at the time of establishment (see \nameref{sec:pestNe}).
Given the heterozygote establishes it will quickly increase to $1/P_\mathrm{est}^-$ copies, implying that fixation effectively occurs when the deterministic trajectory reaches a frequency of $p_f = 1-1/(2 N P_\mathrm{est}^-)$.
Below we will explore scenarios where fixation is expected to occur while the population is at carrying capacity. 
Thus in all cases we have the same effective final allele frequency,
\begin{equation}\label{eq:pf}
p_f = 1 - \frac{1}{2 N(0) P_\mathrm{est}},
\end{equation}
where $P_\mathrm{est}$ is given by Equation \ref{eq:pest} with $\epsilon = 1 - (1 + s h)/(1 +  s)$ and $v$ as given in \nameref{sec:pestNe}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Time to fixation}
\label{sec:fixation_time}

We can estimate the time to fixation in the additive model ($h=1/2$) as the time it takes the deterministic approximation (Equation \ref{eq:ptnt}) to go from the effective initial frequency, $p_0$, to the effective final frequency, $p_f$, which is \citep[equation 5.3.13 in ][]{crow1970introduction}
\begin{equation}\label{eq:tfix}
t_f = \frac{2}{s} \log\left[ \frac{(1-p_0)p_f}{p_0(1-p_f)}\right].
\end{equation}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Backward-time dynamics}
\label{sec:backward_dynamics}

Setting $t = t_f - \tau$ in Equation \ref{eq:ptnt} gives an approximation of the dynamics of allele frequency and population size backward in time,
\begin{equation}\begin{aligned}\label{eq:backwardsptnt}
p^\prime(\tau) &\approx \left(1 + \frac{1 - p_f}{p_f} e^{s \tau/2} \right)^{-1}\\
N^\prime(\tau) &\approx N(0) e^{-d (t_f-\tau)} \left(\frac{1-p_0}{1-p^\prime(\tau)}\right)^{2(1-d)},
\end{aligned}\end{equation}
from the point of fixation, $\tau=0$, to the time of establishment, $\tau=t_f$.
Note that the backward-time allele frequency dynamics do not depend on the initial frequency or $d$ except in determining the maximum value of $\tau$.
Thus the backward-time allele frequency dynamics are expected to follow the same trajectory in rescue as in populations of constant size with the same carrying capacity (such that the final frequency, $p_f$, is the same).
Meanwhile, the backward-time population size dynamics depend heavily on the initial frequency as the timing of the sweep determines the depth and duration of the bottleneck. 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{The structured coalescent}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Event rates}
\label{sec:rescue_backward}

To explore the genetic patterns created by evolutionary rescue we next consider a random sample of chromosomes at the time the beneficial allele fixes.
Focusing on a neutral locus that is recombination distance $r$ from the selected site, we are interested in calculating the rate of coalescence, the rates of recombination and mutation off the selected background, and the rate of migration out of the population.
If our sample of alleles has $k$ distinct ancestors on the selected background $\tau$ generations before fixation, these rates are approximately (table 1 in \citealp{hudson1988coalescent}; equation 16 in \citealp{pennings2006soft2}; see \nameref{sec:derive_coalescent} for derivations)
\begin{equation}\label{eq:coal_rates}
\begin{aligned}
p_{\mathrm{coal}}(k,\tau) &= \binom{k}{2} \frac{1}{2N_e'(\tau) p^\prime(\tau)}\\
p_{\mathrm{rec}}(k,\tau) &= k r [1 - p^\prime(\tau)]\\
p_{\mathrm{mut}}(k,\tau) &= k \frac{u [1 - p^\prime(\tau)]}{p^\prime(\tau)}\\
p_{\mathrm{mig}}(k,\tau) &= k \frac{m}{2N^\prime(\tau)p^\prime(\tau)},
\end{aligned}
\end{equation} 
where $N_e'(\tau)$ is the effective population size $\tau$ generations before fixation.

With slow changes in population size, $N(t-1)\approx N(t)$ and the mean number of gametes contributed to the next generation by each diploid individual in the current generation is $2$.
The inbreeding and variance effective population size, $N_e(t)$, is then roughly $(4N(t) - 2)/(\sigma^2+2)$ \citep[equation 7.6.4.3 in][]{crow1970introduction}, where $\sigma^2$ is the variance in the per capita number of gametes contributed to the next generation.
Therefore, in a large population, $N(t)\gg1$, the ratio of the effective size to the census size is roughly $N_e(t)/N(t) \approx 4/(2+\sigma^2)$, where $\sigma^2$ depends on the particular lifecycle (e.g., see \nameref{sec:pestNe}). 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Event timing}

We now use the instantaneous event rates (Equation \ref{eq:coal_rates}) to calculate the probability that the most recent event is $\tau$ generations before fixation and is either coalescence, recombination, mutation, or migration. 
Letting $i,j\in\{\mathrm{coal, rec, mut, mig}\}$, the probability that $i$ is the most recent event, and occurs $\tau$ generations before fixation, is \citep[c.f., equation 6 in][]{pennings2006soft}
\begin{equation}\label{eq:Pikt}
\begin{aligned}
P_i(k,\tau) &= p_{i}(k,\tau) \left( \prod_{j\neq i} [1 - p_{j}(k,\tau)] \right) \prod_{l=0}^{\tau-1} \left( \prod_{j} [1 - p_{j}(k,l)] \right)\\
&\approx p_{i}(k,\tau) \exp \left( -\sum_{j} \int_{0}^{\tau} p_{j}(k,l) \mathrm{d}l \right),
\end{aligned}
\end{equation}
i.e., the waiting time for an inhomogeneous exponential random variable.
The approximation assumes the $p_i(k,\tau)$ are small enough such that at most one event happens each generation, with small probability, and the changes in the $p_i(k,\tau)$ from one generation to the next are small enough that we can approximate a sum across $\tau$ with an integral.
As a technical aside, to speed computation we analytically solve the integrals and evaluate Equation \ref{eq:Pikt} under the assumption of weak selection and exponential population growth (Equation \ref{eq:ptnt}).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Genetic signatures at linked neutral loci}
\label{sec:rescue_signatures}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Genetic diversity}
\label{sec:rescue_heterozygosity}

One classic pattern of genetic variation produced by a selective sweep is a dip in genetic diversity around the selected site \citep{smith1974hitch,kaplan1989hitchhiking}. 
Here we consider the average number of nucleotide differences between two randomly sampled sequences, $\pi$ \citep{tajima1983evolutionary}, focusing on sequences of length 1 (i.e., heterozygosity).

We first consider our expectation for $\pi$ at a site that is far enough away from the selected site to be unaffected by hitchhiking, which will provide us with an expectation for  genome-wide average diversity.
While not directly impacted by the sweep, these loci are indirectly impacted as the sweep dictates the severity of the population bottleneck.
Our expectation for $\pi$ at such a site in a population of constant effective size, $N_e$, is simply $\theta = 4 N_e U$ \citep{watterson1975number}, with $U$ the per base per generation mutation rate at neutral loci.
Ignoring neutral mutation input during the sweep (which will cause us to underestimate diversity), the $\pi$ at a sufficiently loosely linked site is this neutral expectation times the probability a sample of size two does not coalesce during the period of interest (in our case, from the time we take the sample at the time of fixation, $\tau=0$, until the time of environmental change, $\tau=t_f$).
This is (c.f., equation 4 in \citealp{slatkin1991pairwise} and equation 7 in \citealp{griffiths1994sampling})
\begin{equation}\label{eq:EH_rescue_loose}
\mathbb{E}[\pi | \mathrm{unlinked}]\approx \theta \exp \left( - \int_0^{t_f} \frac{1}{2N_e'(\tau)} \mathrm{d}\tau \right). 
\end{equation}
In a population of constant size we simply use $\mathbb{E}[\pi | \mathrm{unlinked}]=\theta$, which is more accurate as it incorporates neutral mutations occurring during the sweep.

We next consider sites that are closer to the selected locus, such that they are directly affected by the selective sweep through hitchhiking.
To keep the analysis simple, we assume that if recombination or mutation moves one of the sampled alleles to the ancestral background before the two samples coalesce then it is as if both samples were on the ancestral background from the start and therefore coalesce with each other as if they were at an unlinked locus (Equation \ref{eq:EH_rescue_loose}).
This assumption therefore ignores the time it takes for the two samples to arrive on the ancestral background and is therefore expected to overestimate coalescence rates.
There is also the possibility of no events occurring in the history of the sample during the sweep.
Then, if the sweep arose from mutation or migration or from a single copy of the beneficial allele ($\kappa=1$), the sample must coalesce (i.e., we observe a hard sweep). 
If we instead start with more than one beneficial copy, $\kappa>1$, then under the assumption that each copy of the beneficial allele in the standing variance has a recent and unique mutational origin \citep[which we will assume throughout for simplicity, noting this effectively implies the beneficial allele was sufficiently deleterious before the environmental change; c.f.,][]{prezeworski2005signature}, the samples are independent draws from a neutral population (i.e., the samples to do not coalesce and we observe a soft sweep).
Ignoring neutral mutations during the sweep, a simple approximation for diversity at any location in the genome is then
\begin{equation}\label{eq:EH_rescue}
\begin{aligned}
\mathbb{E}[\pi ] \approx P_\mathrm{off}(2,t_f) \mathbb{E}[\pi | \mathrm{unlinked}] + P_\emptyset(2,t_f) \delta_{\kappa>1} \theta,
\end{aligned}
\end{equation}
where $P_\mathrm{off}(k,t_f) = \int_0^{t_f} [P_\mathrm{rec}(k,\tau) + P_\mathrm{mut}(k,\tau)] \mathrm{d}\tau$ is the probability of recombination or mutation before coalescence during the sweep, $P_\emptyset(k, t_f) = \exp \left( -\sum_{j} \int_{0}^{t_f} p_{j}(k,\tau) \mathrm{d}\tau \right)$ is the probability that no events have occurred in the history of the sample during the sweep ($j\in\{\mathrm{rec},\mathrm{mut},\mathrm{coal}\}$), and $\delta_{\kappa>1}$ is 1 if the sweep arose from more than one copy of the beneficial allele ($\kappa>1$) and 0 otherwise.
Equation \ref{eq:EH_rescue} is a function of recombination distance, $r$, and approaches $\mathbb{E}[\pi | \mathrm{unlinked}]$ as $r$ gets large.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Tajima's \textit{D}}
\label{sec:tajimasD}

As a second genetic signature at linked neutral sites we consider Tajima's $D$ statistic \citep{tajima1989statistical}, which measures the excess (positive $D$) or deficiency (negative $D$) of intermediate frequency polymorphisms relative to the standard neutral model. 
Quantitative predictions of Tajima's $D$ require one to consider samples of size greater than 2, which quickly becomes complicated with selection and non-equilibrial demography.
Instead, here we discuss the expected qualitative patterns based on intuition from the analysis presented above.

First, hard selective sweeps tend to produce star-like gene genealogies, with most samples coalescing near the beginning of the sweep and recombination allowing a few samples to coalesce much further back in time \citep{kaplan1989hitchhiking}.
Hard sweeps therefore produce an excess of low and high frequency polymorphisms \citep[][p.\ 120]{wakeley2009coalescent}, leading to negative $D$ \citep{braverman1995hitchhiking}.
Larger selection coefficients create more star-like the genealogies (as there is then less time for recombination off the sweep) and thus more negative $D$ when conditioned on a hard sweep.
However, with sufficient standing genetic variation or rates of recurrent mutation or migration, larger selection coefficients will tend to cause softer selective sweeps as they increase the probability any one copy establishes (Equation \ref{eq:pest}).
Soft selective sweeps allow some samples to coalesce further back in time, before the start of the sweep, even at the selected site.
Soft sweeps therefore tend to have less effect on neutral genealogies and hence on $D$, although sufficiently soft sweeps can actually cause positive $D$, by allowing intermediate-sized groups of samples to descend from different ancestors containing the beneficial allele \citep{pennings2006soft}.

As linkage to the selected site decreases so too does this skew in genealogies.
In the case of a constant population size, $D$ should asymptote to the neutral expectation of zero.
In the case of rescue, however, the bottleneck will cause an excess of intermediate frequency polymorphisms \citep[][p.\ 120]{wakeley2009coalescent}, and therefore $D$ should asymptote at some positive value (more positive with more severe bottlenecks).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Simulations}
\label{sec:simulations}

The simulation details are described in full in \nameref{sec:simdetails}.
Briefly, the lifecycle described in \nameref{sec:lifecycle}, with the addition of a hard carrying capacity at $N(0)$, was simulated in \texttt{SLiM} 3 \citep{Haller2019slim} with tree-sequence recording \citep{Haller2019tree}.
We simulated a 20 Mb segment of a chromosome, with all but one of the center loci neutral, with a per base recombination rate of $r_{bp}=2\times10^{-8}$ and per base mutation rate at neutral loci of $U=6\times10^{-9}$ \citep[both inspired by \textit{Drosophila} estimates;][]{mackay2012drosophila,haag2007direct}.
A population was considered rescued when the beneficial mutation was fixed and the population size had recovered to its initial size, $N(0)$.
Pairwise nucleotide diversity, Tajima's $D$, site frequency spectra, and linkage disequilibrium were calculated directly from the tree sequences using \texttt{msprime} \citep{kelleher2016efficient} and \texttt{tskit} \cite[][]{kelleher2018efficient}.
We chose to simulate a population with initial census size $N(0)=10^4$ (giving initial effective size $N_e(0)\approx4 N(0)/7$) declining at rate $d=0.05$, where the beneficial allele had selection coefficient $s=0.13$ or $s=0.20$.
This describes a relatively small population that is expected to go extinct in $\approx 200$ generations in the absence of a sweep at a locus under strong selection (as might be expected following a severe environmental change).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Data availability statement}
\label{sec:data}

Code used to derive and plot all results (Python scripts and Mathematica notebook) is available at \url{https://github.com/mmosmond/rescueCoalescent.git}.
File S1 refers to the Mathematica notebook, which is also provided in freely accessible forms (CDF and PDF).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Results}
\label{sec:results}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Rescue from standing genetic variation (SGV)}
\label{sec:rescue_forward_SGV}

We first consider rescue arising from genetic variation that is present at the time of the environmental change and ignore further mutations and migration.
To avoid complicating the presentation we assume the initial number of beneficial alleles is given; treating the initial number as a random variable requires conditioning the distribution on a successful sweep \citep[][]{hermisson2017soft}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{The probability of rescue and soft selective sweeps}

Given there are initially $\kappa\ll N(0)$ copies of the beneficial allele, the number that establish is roughly binomially distributed with $\kappa$ trials and success probability $P_\mathrm{est}$.
The probability of rescue is the probability that at least one establishes \citep[c.f., equation 2 in][]{orr2014population},
\begin{equation}
P_\mathrm{rescue}^\mathrm{SGV} = 1-(1-P_\mathrm{est})^\kappa.
\end{equation}
Conditioning on rescue, the expected number of establishing copies is
\begin{equation}\label{eq:ENestSGV}
n^\mathrm{SGV}_\mathrm{rescue} = \frac{\kappa P_\mathrm{est}}{P^\mathrm{SGV}_\mathrm{rescue}}.
\end{equation}
Note that when the expected number of establishing copies, $\kappa P_\mathrm{est}$, is small, the probability of rescue is roughly $\kappa P_\mathrm{est}$.
Equation \ref{eq:ENestSGV} then implies we expect only one of the initial copies to establish, i.e., we expect a hard selective sweep.
For larger values of $\kappa P_\mathrm{est}$ rescue will often occur by a soft selective sweep \citep{hermisson2005soft}, where multiple of the initial copies establish.
The probability that multiple copies establish is the probability of rescue minus the probability of a hard sweep, $P^\mathrm{SGV}_\mathrm{rescue} - \kappa P_\mathrm{est} (1-P_\mathrm{est})^{\kappa-1}$.
Given rescue occurs, the probability it is due to a soft sweep can therefore be written as
\begin{equation}\label{eq:PsoftSGV}
P_{\mathrm{soft}|\mathrm{rescue}}^\mathrm{SGV} = 1 -n^\mathrm{SGV}_\mathrm{rescue} \frac{1-P_\mathrm{rescue}^\mathrm{SGV}}{1-P_\mathrm{est}}.
\end{equation}
As expected, the probability of a soft sweep given rescue is 0 when we start with a single copy, $\kappa=1$, and approaches 1 as the number of copies becomes large.
Between these two extremes we find that Equation \ref{eq:PsoftSGV} provides reasonable estimates for small $d$ or large $s$ but to underestimate the probability of a soft sweep otherwise (Figure \ref{fig:PsoftSGV}A), when beneficial alleles can persist at low numbers long enough to establish with some non-negligible probability of experiencing some selection as homozygotes (and thus increasing their probability of establishment).
Using the probability of establishment observed in the simulations corrects this error (see File S1).
Because the expected number of copies that establishes given rescue (Equation \ref{eq:ENestSGV}) is nearly independent of $P_\mathrm{est}$ (as long as it is small), this equation more closely approximates the results from simulations (Figure \ref{fig:PsoftSGV}B).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}
\node[] (A) {\includegraphics[width=0.49\linewidth]{PsoftSGV.pdf}};
\node[below = -1cm of A] (B) {\includegraphics[width=0.49\linewidth]{NumberEstSGV.pdf}};
\node[below = -1cm of A, xshift = -0.75cm, darkred, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont\textbf{constant $\bm{N}$}};
\node[right = 0cm of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont\textbf{evol.\ rescue}};
\draw[very thick] (-1,-2.5) -- (-0.25,-2.5); 
\draw[fill=black] (0,-2.5) circle (0.05cm);
\node at (0.75, -2.475) {$s=0.2$}; 
\draw[very thick] (-1,-2.9) -- (-0.75,-2.9); \draw[very thick] (-0.5,-2.9) -- (-0.25,-2.9); 
\draw[very thick] (0,-2.9) circle (0.05cm);
\node at (0.84, -2.875) {$s=0.13$}; 
\end{tikzpicture}
\caption{
Soft sweeps from standing genetic variation.
(A) The probability that more than one initial copy of the beneficial allele has descendants at the time of fixation in rescue (blue; $d=0.05$) or in a population of roughly constant size (red; $d=0$) as a function of the initial number of copies of the beneficial allele, $\kappa$.
(B) The expected number of initial copies of the beneficial allele that have descendants at the time of fixation.
The curves are Equations \ref{eq:PsoftSGV} (panel A) and \ref{eq:ENestSGV} (panel B).
Each point is based on 100 replicate simulations where a sweep (and population recovery) was observed.
Error bars in (B) are standard errors.
Parameters: $N(0)=10^4$.
}%
\label{fig:PsoftSGV}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Effective initial allele frequency and the backward-time dynamics}

As each of the $n^\mathrm{SGV}_\mathrm{rescue}$ expected establishing copies given rescue is expected to rapidly increase to a frequency of $1/P_\mathrm{est}$ (Section \nameref{sec:initial_frequency}), it is as if the sweep began deterministically (Equation \ref{eq:ptnt}) with initial frequency \citep[c.f., equation S1.4 in][]{orr2014population}
\begin{equation}\label{eq:p0sgv}
p_{0|\mathrm{rescue}}^\mathrm{SGV} = \frac{n^\mathrm{SGV}_\mathrm{rescue}}{2N(0)P_\mathrm{est}} = \frac{\kappa}{2N(0)} \frac{1}{P_{\mathrm{rescue}}^{\mathrm{SGV}}}.
\end{equation}
This same conditioning applies in a population of constant size (i.e., $d=0$).
As the decline rate, $d$, increases the probability a copy establishes, $P_\mathrm{est}$, and hence the probability of rescue, $P_\mathrm{rescue}^\mathrm{SGV}$, declines, making the conditioning stronger (i.e., the effective initial allele frequency is much greater than the true initial allele frequency).
This causes selective sweeps to get started faster as $d$ increases, implying that, for a given $s$, rescue sweeps are shorter than those in populations of constant size.

Using $p_{0|\mathrm{rescue}}^\mathrm{SGV}$ as $p_0$ in Equations \ref{eq:tfix} and \ref{eq:backwardsptnt} gives our prediction of the backward-time dynamics.
As shown in Figure \ref{fig:rescueSGV_dynamics}, these semi-deterministic predictions do a reasonably good job of describing the simulation results.
Looking closer, we find that our predictions do an excellent job of approximating the mean allele frequency as it departs from 1, but typically begin to underestimate the mean allele frequency as frequencies drop lower (see File S1).
Consequently, we tend to underestimate the mean time to fixation (compare arrows and stars in Figure \ref{fig:rescueSGV_dynamics}) and overestimate mean population size.
This error arises even under a true branching process (e.g., under clonal reproduction) in a population of constant size, and becomes worse as selection becomes weaker (see File S1).
The reason for this error is that one minus the effective final allele frequency, $1-p_f = q_f$ is exponentially distributed with expectation $1/(2N(0)P_\mathrm{est}^-)$ \citep{martin2015simple}, where $P_\mathrm{est}^-$ is the backward-time establishment probability of the heterozygote in a population of mutant homozygotes and we have assumed fixation occurs when the population size is $N(0)$ (which is expected to be true in all our numerical examples).  
The variance in the effective final allele frequency therefore increases like $1/P_\mathrm{est}^2 \sim 1/s^2$ as $s$ declines, meaning that under weaker selection there is much more variance in the time spent in the stochastic phase near fixation.
Combining this distribution of waiting times to leave the stochastic phase with the non-linearities of the subsequent deterministic phase causes our semi-deterministic approximation to generally underestimate allele frequency.
Despite this error we forge ahead and use our simple semi-deterministic approximations to predict rates of coalescence (which are in general non-linear functions of these approximations; Equations \ref{eq:coal_rates}-\ref{eq:Pikt}).

Before moving on to the coalescent, a number of biological insights can be gleaned from Figure \ref{fig:rescueSGV_dynamics} that will help explain downstream results:
1) the backward-time allele frequency dynamics are the same in rescue as they are in populations of constant size since they only depend on $p_f$ (i.e., they do not depend on $d$; Equation \ref{eq:pf}),
2) fixation times are slightly shorter in rescue as, forward-in-time, those sweeps tend to get started faster (i.e., the effective initial frequency increases with $d$; Equation \ref{eq:p0sgv}),
3) weaker selection causes larger population bottlenecks as the deterministic portion of the sweep is then slower and a given frequency change has less impact on population mean fitness (Equation \ref{eq:ptnt}), and
4) both the allele frequency and population size dynamics depend only weakly on the initial number of mutants, despite large differences in the probability of rescue, because sweeps that are less likely to occur get started faster (Equation \ref{eq:p0sgv}).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htbp]
\centering
\begin{tikzpicture}
\node[] (A1) {\includegraphics[width=0.49\linewidth]{pdynamics_rescue_s0.2_k100_u0_m0.pdf}};
\node[below = -0.9cm of A1] (A2) {\includegraphics[width=0.49\linewidth]{ndynamics_rescue_s0.2_k100_u0_m0.pdf}};
\node[right = 0cm of A1] (B1) {\includegraphics[width=0.49\linewidth]{pdynamics_rescue_s0.2_k10_u0_m0.pdf}};
\node[right = 0cm of A2] (B2) {\includegraphics[width=0.49\linewidth]{ndynamics_rescue_s0.2_k10_u0_m0.pdf}};
\node[below = 0cm of A2] (C1) {\includegraphics[width=0.49\linewidth]{pdynamics_rescue_s0.13_k100_u0_m0.pdf}};
\node[below = -0.9cm of C1] (C2) {\includegraphics[width=0.49\linewidth]{ndynamics_rescue_s0.13_k100_u0_m0.pdf}};
\node[right = 0cm of C1] (D1) {\includegraphics[width=0.49\linewidth]{pdynamics_rescue_s0.13_k10_u0_m0.pdf}};
\node[right = 0cm of C2] (D2) {\includegraphics[width=0.49\linewidth]{ndynamics_rescue_s0.13_k10_u0_m0.pdf}};
\node[right = 0.5 of A1, yshift = -2.5cm, rotate = 90, draw, rectangle, rounded corners, thick, fill=gray!50] (s) {\large$\bm{s=0.20}$};
\node[right = 0.5 of C1, yshift = -2.5cm, rotate = 90, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{s=0.13}$};
\node[below = -0.6cm of A2, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] (k) {\large$\bm{\kappa=100}$};
\node[below = -0.6cm of B2, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{\kappa=10}$};
\node[right = 2cm of k, darkred, yshift = 0.25cm, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont\textbf{constant $\bm{N}$}};
\node[below = 0cm of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont\textbf{evol.\ rescue}};
\node (pt) at ($(A1) + (-1cm,-1cm)$) {\footnotesize\fontfamily{phv}\selectfont mean fixation times};
\draw (pt) -- ($(pt)+(1.5,-0.4)$);
\node (nt) at ($(A1) + (0.1cm,0cm)$) {\footnotesize\fontfamily{phv}\selectfont predicted fixation times};
\draw (nt) -- ($(nt)+(0.85,-1.7)$);
\end{tikzpicture}
\caption{
Allele frequency, $p^\prime(\tau)$, and population size, $N^\prime(\tau)$, at generation $\tau$ before fixation during a selective sweep from standing genetic variation in evolutionary rescue (blue; $d=0.05$) and in a population of roughly constant size (red; $d=0$).
The thick solid curves are analytic approximations (Equation \ref{eq:backwardsptnt}), using $p_0 = p_{0|\mathrm{rescue}}^\mathrm{SGV}$ (Equation \ref{eq:p0sgv}) as the initial frequency and $p_f$ (Equation \ref{eq:pf}) as the final allele frequency.
The thin curves are 100 replicate simulations where a sweep (and population recovery) was observed.
The stars show the predicted time to fixation (Equation \ref{eq:tfix}).
The arrows show the mean time to fixation observed in simulations. 
}%
\label{fig:rescueSGV_dynamics}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{The structured coalescent}

For a linked neutral allele in a sample of size 2 Figure \ref{fig:rescueSGV_timing} compares the predicted probability of recombination off the selective sweep (opaque dashed) and coalescence on the beneficial background (opaque solid) with the mean probabilities given the allele frequency and population size dynamics observed in simulations (transparent curves).
We find that our approximations do relatively well near fixation, where our predictions of allele frequency are better (Figure \ref{fig:rescueSGV_dynamics}), but that our underestimates of allele frequency at later times cause us to generally overestimate rates of recombination and coalescence closer to the beginning of the sweep.
Despite this error we capture the qualitative dynamics and can draw out a number of interesting biological consequences: 
1) the timing of the bottleneck during rescue pushes coalescent times towards the present, so that the distributions of coalescence and recombination times overlap more,
2) the bottleneck also increases the overall probability of coalescence in rescue, which reduces the probability of recombination off the sweep (compare areas under dashed curves), and
3) the difference in the structured coalescent between rescue and a population of constant size is larger under weaker selection.
This latter point nicely illustrates the coupling between demography and evolution in rescue; while weaker selection creates a slower sweep and hence more time for recombination off the beneficial background, it also slows population recovery in the case of rescue, leading to longer and deeper bottlenecks such that increased coalescence counteracts the additional time provided for recombination.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}
\node[] (A1) {\includegraphics[width=0.49\linewidth]{coalescent_rescue_s0.2_k100_u0_m0.pdf}};
\node[right = 0cm of A1] (B1) {\includegraphics[width=0.49\linewidth]{coalescent_rescue_s0.2_k10_u0_m0.pdf}};
\node[below = 0cm of A1] (C1) {\includegraphics[width=0.49\linewidth]{coalescent_rescue_s0.13_k100_u0_m0.pdf}};
\node[right = 0cm of C1] (D1) {\includegraphics[width=0.49\linewidth]{coalescent_rescue_s0.13_k10_u0_m0.pdf}};
\node[right = 0.5 of A1, yshift = -0.5cm, rotate = 90, draw, rectangle, rounded corners, thick, fill=gray!50] (s) {\large$\bm{s=0.20}$};
\node[right = 0.5 of C1, yshift = -0.5cm, rotate = 90, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{s=0.13}$};
\node[below = -0.6cm of A1, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] (k) {\large$\bm{\kappa=100}$};
\node[below = -0.6cm of B1, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{\kappa=10}$};
\node[right = 2cm of k, darkred, yshift = 0.25cm, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont\textbf{constant $\bm{N}$}};
\node[below = 0cm of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont\textbf{evol.\ rescue}};
\node (pt) at ($(A1) + (-1cm,-0cm)$) {\footnotesize\fontfamily{phv}\selectfont coalescence};
\draw (pt) -- ($(pt)+(2,-1)$);
\node (nt) at ($(A1) + (0cm,1cm)$) {\footnotesize\fontfamily{phv}\selectfont recombination};
\draw (nt) -- ($(nt)+(1.4,-1/2)$);
\end{tikzpicture}
\caption{
The timing of events in the structured coalescent (Equation \ref{eq:Pikt}) for a sample of size $k=2$ at a linked neutral locus ($r=0.01$) during a selective sweep from standing genetic variation in evolutionary rescue (blue; $d=0.05$) or in a population of roughly constant size (red; $d=0$).
The opaque curves use the analytic expressions for allele frequency and population size (Equation \ref{eq:backwardsptnt}) while the transparent curves show the mean probabilities given the observed allele frequency and population size dynamics in 100 replicate simulations (these become more variable with time as less replicates remain polymorphic).
}%
\label{fig:rescueSGV_timing}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Genetic diversity}

Figure \ref{fig:rescueSGV_heterozygosity_relative} compares our predictions of relative pairwise diversity, $\mathbb{E}[\pi]/\mathbb{E}[\pi|\mathrm{unlinked}]$ (Equations \ref{eq:EH_rescue_loose}-\ref{eq:EH_rescue}), against simulations, $\pi/\overline{\pi}$.
The figure shows that despite the errors in the predictions of the dynamics above (Figure \ref{fig:rescueSGV_dynamics}-\ref{fig:rescueSGV_timing}) our approximations do manage to capture the relative rates of recombination and coalescence.
Two main conclusions emerge:
1) rescue tends to deepen dips in diversity when soft sweeps are possible, i.e., it hardens soft sweeps (the bottleneck increases the probability of coalescence and thus decreases $P_\emptyset(2,T)$ at the selected site, $r=0$; Equation \ref{eq:EH_rescue}), and
2) rescue generally produces wider dips in diversity due to excess, and earlier, coalescence as well shorter sweeps (the magnitude and timing of the bottleneck, as well as lower establishment probabilities, decreases $P_\mathrm{off}(2,T)$ at a given $r>0$; Equation \ref{eq:EH_rescue}).
Note that when the probability of rescue is very small (Figure \ref{fig:rescueSGV_heterozygosity_relative}D), we predict the diversity at the selected site to actually be higher under rescue than in a population of constant size.
This is driven by our prediction of fixation time; low probabilities of rescue imply high effective initial allele frequencies, leaving less time for coalescence to occur (compare solid curves in Figure \ref{fig:rescueSGV_timing}D).
This is prediction is not, however, borne out in the simulations presented here, as we tend to overestimate coalescence in populations of constant size case and underestimate it in rescue (Figure \ref{fig:rescueSGV_timing}D).

If the population was sampled both before and after the selective sweep (or we have very good estimates of its mutation rate and long-term effective population size), the absolute amount of pairwise diversity contains additional information.
Figure \ref{fig:rescueSGV_heterozygosity_empirical} compares our predictions of $\pi$ against simulations, and shows that the bottleneck during rescue decreases background levels of diversity. 
This decreases is more evident under weaker selection, where bottlenecks are more pronounced.
While our predictions qualitatively match the mean simulation results, the tendency of our semi-deterministic approximations to underestimate allele frequencies and thus overestimate harmonic mean population sizes during rescue (Figures \ref{fig:rescueSGV_dynamics}) causes us to generally overestimate background diversity in these cases.
To correct for this, we can replace our prediction for background diversity, $\mathbb{E}[\pi | \mathrm{unlinked}]$ (Equation \ref{eq:EH_rescue_loose}), in the rescue scenario with the observed genome-wide average diversity (dashed curves in Figure \ref{fig:rescueSGV_heterozygosity_empirical}.
A very similar result is achieved by using observed effective population sizes in Equation \ref{eq:EH_rescue_loose} (results not shown). 
Using the observed mean diversity level may be justified by the fact that genome-wide diversity can be measured directly from data and is highly variable across populations \citep{tajima1983evolutionary}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}
\node[] (A) {\includegraphics[width=0.49\linewidth]{EH_rescue_s0.2_k100_u0_m0_relative.pdf}};
\node[right = 0 of A] (B) {\includegraphics[width=0.49\linewidth]{EH_rescue_s0.2_k10_u0_m0_relative.pdf}};
\node[below = 0 of A] (C) {\includegraphics[width=0.49\linewidth]{EH_rescue_s0.13_k100_u0_m0_relative.pdf}};
\node[right = 0 of C] (D) {\includegraphics[width=0.49\linewidth]{EH_rescue_s0.13_k10_u0_m0_relative.pdf}};
\node[right = 0 of A, xshift = 0.6cm, yshift = -0.5cm, draw, rectangle, rounded corners, thick, fill=gray!50, rotate = 90] (s) {\large$\bm{s=0.20}$};
\node[right = 0 of C,  xshift = 0.6cm, yshift = -0.5cm, draw, rectangle, rounded corners, thick, fill=gray!50, rotate = 90] {\large$\bm{s=0.13}$};
\node[below = -2/3 of A, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] (k) {\large$\bm{\kappa=100}$};
\node[below = -2/3 of B, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{\kappa=10}$};
\node[right = 2cm of k, darkred, yshift = 0.25cm, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont \textbf{constant $\bm{N}$}};
\node[below = 0 of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont \textbf{evol.\ rescue}};
\end{tikzpicture}
\caption{
Relative mean pairwise diversity, $\pi/\overline{\pi}$, after a selective sweep from standing genetic variation during evolutionary rescue (blue; $d=0.05$) or in a population of roughly constant size (red; $d=0$).
The thickest curves are $\mathbb{E}[\pi]/\mathbb{E}[\pi|\mathrm{unlinked}]$ (Equations \ref{eq:EH_rescue_loose}-\ref{eq:EH_rescue}).
The thinnest curves are 100 replicate simulations (rescue only for clarity) and the slightly thicker curves are simulation means (often obscured by prediction).
Parameters: $N(0)=10^4$.
}%
\label{fig:rescueSGV_heterozygosity_relative}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Tajima's $D$}

Figure \ref{fig:rescueSGV_tajimasD} shows the Tajima's $D$ values observed in simulations.
There are two main take-aways:
1) the bottleneck during rescue consistently causes positive background $D$, and
2) when there is some possibility of a soft sweep ($\kappa>1$) rescue tends to harden the sweep and thus produce lower values of $D$ at the selected site.
Together these patterns cause rescue to ``stretch out" (e.g., panel B) or even invert (e.g., panel C) the pattern of $D$ observed in populations of constant size.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}
\node[] (A) {\includegraphics[width=0.49\linewidth]{TajimasD_rescue_s0.2_k100_u0_m0.pdf}};
\node[right =0 of A] (B) {\includegraphics[width=0.49\linewidth]{TajimasD_rescue_s0.2_k10_u0_m0.pdf}};
\node[below =0 of A] (C) {\includegraphics[width=0.49\linewidth]{TajimasD_rescue_s0.13_k100_u0_m0.pdf}};
\node[right =0 of C] (D) {\includegraphics[width=0.49\linewidth]{TajimasD_rescue_s0.13_k10_u0_m0.pdf}};
\node[right = 0 of A, xshift = 0.6cm, yshift = -0.5cm, draw, rectangle, rounded corners, thick, fill=gray!50, rotate = 90] (s) {\large$\bm{s=0.20}$};
\node[right = 0 of C,  xshift = 0.6cm, yshift = -0.5cm, draw, rectangle, rounded corners, thick, fill=gray!50, rotate = 90] {\large$\bm{s=0.13}$};
\node[below = -2/3 of A, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] (k) {\large$\bm{\kappa=100}$};
\node[below = -2/3 of B, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{\kappa=10}$};
\node[right = 2cm of k, darkred, yshift = 0.25cm, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont \textbf{constant $\bm{N}$}};
\node[below = 0 of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont \textbf{evol.\ rescue}};
\end{tikzpicture}
\caption{
Tajima's $D$ after a selective sweep from standing genetic variation in evolutionary rescue (blue; $d=0.05$) or in a population of roughly constant size (red; $d=0$).
Thin curves show 100 replicate simulations (rescue only for clarity) and thicker curves show simulation means.
Parameters: $N(0)=10^4$.
}%
\label{fig:rescueSGV_tajimasD}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{A simultaneous, but independent, sweep and bottleneck}

Above we have compared the genetic signatures of rescue to those of sweeps in populations of constant size. 
Two of key differences we have identified, lower absolute diversity and positive Tajima's $D$ under rescue, arise because of the population bottleneck.
This suggests that it will be yet harder to distinguish rescue from a sweep that occurs during a bottleneck, where the sweep and bottleneck are coincident but independent.
Taking this argument to its logical limit, if the population size dynamics during an externally forced bottleneck are exactly like those expected under rescue, then the expected genetic signatures will be identical.
As a middle-ground, we can compare an instance of rescue that has an expected effective population size of $N_e$ during its sweep to a population that is bottlenecked to a constant effective population size $N_e$ during its sweep.
Then, under the same allele frequency dynamics (which will not be true as population size and decline rate affect the effective initial and final allele frequencies), both populations would have similar mean diversity and Tajima's $D$ values.

In File S1 we show that populations bottlenecked in such a way have similar levels of relative diversity at the selected site as do populations that are not bottlenecked (i.e., like the red curve at 0cM in Figure \ref{fig:rescueSGV_heterozygosity_relative}), meaning that rescue causes deeper dips in diversity than a coincident sweep and bottleneck.
This is because the bottleneck scenario assumes smaller population sizes at the beginning and end of the sweep than under rescue, causing a higher effective initial frequency and a lower effective final frequency, which shortens the length of the sweep, $t_f$, and thus increases the probability that no coalescent events occur during the sweep, $P_\emptyset(k,t_f)$.
These shorter sweeps also mean that bottlenecked populations also have higher mean absolute diversity than populations that have been rescued (i.e., like the red curves in Figure \ref{fig:rescueSGV_heterozygosity_empirical}).
In contrast, both rescue and the bottleneck similarly slow the recovery of relative diversity as we move away from the selected site, causing their dips in diversity to have similar width (i.e, like the blue curve in Figure \ref{fig:rescueSGV_heterozygosity_relative}).
Rescue and the bottleneck slow the rate at which diversity recovers with recombination because both increase rates of coalescence early in the sweep, preventing recombination off and thus lowering $P_\mathrm{off}(k,t_f)$.
The shape of Tajima's $D$ values across the genome are much the same after a sweep in a bottlenecked population as they are after a sweep in a population of constant size (i.e., like the red curves in Figure \ref{fig:rescueSGV_tajimasD}).
However, the bottleneck shifts these values up so that the background levels closely match those under rescue (i.e., they have intercepts like the blue curves in Figure \ref{fig:rescueSGV_tajimasD}).
In summary: for a given strength of selection and effective population size during a sweep, rescue produces 1) deeper dips in diversity, 2) lower absolute diversity genome-wide, and 3) a greater range of Tajima's $D$ values.

To explore whether there is additional signal to be gained from the different timings of coalescence and recombination in a population that was bottlenecked versus one that was rescued, we have also looked at the full site frequency spectra arising from simulations \citep[of which pairwise diversity and Tajima's $D$ are summaries;][p.\ 116]{wakeley2009coalescent}.
Figure \ref{fig:rescueSGV_SFS} shows the site frequency spectra at three recombination distances for a scenario where relative diversity and Tajima's $D$ values are essentially the same in rescue as they are in the bottlenecked populations ($s=0.2$, $\kappa=10$).
At sites far from the selected site the entire spectra are very similar in the two populations, both documenting a recent drop in population size.
As we move towards the selected site it appears that rescue may cause a slightly larger deficit of intermediate frequency alleles and a slightly larger excess of high frequency alleles.
By the time we reach the selected site the spectra again become quite similar, although with more noise as there are then less polymorphic sites.
Thus the strongest signal of rescue vs.\ a bottlenecked sweep is provided at sites an intermediate distance from the selected site, where the relative timings and strengths of coalescence and recombination have the most impact.
Here for instance, the timing and probability of coalescence at 1cM is nearly identical in rescue vs.\ the bottleneck, but the probability of recombination is slightly reduced and tends to occur slightly later under rescue. 
This reduction in recombination is predicted to create phylogenies that are more star-like, which may explain much of the deficit of intermediate frequency mutations (and the slightly more negative value of Tajima's $D$ at the shoulders of the selected site under rescue).
The fact that recombination is also delayed and thus overlaps more with coalescence, in rescue relative to bottlenecks, may help explain the slightly flatter distribution across low to high frequency mutations, as the overlap creates more variance in the number of coalescence events that occur prior to the lineage recombining off the sweep \citep[similar to the effect of recurrent mutation;][]{pennings2006soft}.
Note that if we instead compared site frequency spectra in windows with similar absolute genetic diversity \citep[as in][]{kim2010signatures}, differences between the three models would be more apparent.
Thus a combination of summary statistics may help differentiate rescue from a simultaneous bottleneck and sweep.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}
\node[] (A) {\includegraphics[width=\linewidth]{SFS_rescue_s0.2_k10_bottle.pdf}};
\node[above = 0 of A, darkyellow, draw, rectangle, rounded corners, thick, fill=darkyellow!25] (N) {\fontfamily{phv}\selectfont \textbf{bottleneck}};
\node[left = 0 of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont \textbf{evol.\ rescue}};
\node[right=0 of N, darkred, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont \textbf{constant $N$}};
\end{tikzpicture}
\caption{
Site frequency spectra at three distances from the selected site after a selective sweep from standing genetic variation in evolutionary rescue (blue; $N(0)=10^4$, $d=0.05$), in a bottlenecked population with similar short-term effective population size similar to that in rescue (yellow; $N(0)=2945$, $d=0$), or in a population of constant size (red; $N(0)=10^4$, $d=0$).
Blue, yellow, and red curves are mean values from 100 replicate simulations.
The black curve is the neutral expectation, $4 N_e U / x$, where $N_e$ is the long-term effective population size (here $10^4\times 4/7$) and $x$ is the number of copies of the derived allele.
Parameters: $s=0.2$, $\kappa=10$. 
}%
\label{fig:rescueSGV_SFS}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

Site frequency spectra (and summaries of them) are characteristics of individual loci.
As a final potential signature of rescue vs.\ a bottlenecked sweep we look at linkage disequilibrium, which captures correlations in the pairwise coalescence times between two loci \citep[][p.\ 236]{wakeley2009coalescent}.
Figure \ref{fig:rescueSGV_LD} shows linkage disequilibrium between neutral loci that are a recombination distance of $r=0.001$ apart, as a function of their distance from the selected site.
As we can see, the sweep causes elevated linkage disequilibrium near the selected site, as expected for sites that were segregating before the sweep \citep{kim2004linkage,mcvean2007structure}.
Meanwhile population bottlenecks tend to increase linkage disequilibrium genome wide, as expected given that bottlenecks decrease mean coalescence times more than they decrease the variance \citep{mcvean2002genealogical}.
The differences in linkage disequilibrium patterns between the rescue and bottleneck scenario are much greater than the differences observed in relative diversity, Tajima's $D$, or the full site frequency spectrum. 
The differences in linkage disequilibrium do, however, largely mirror those observed in absolute diversity, which can largely be explained by the shorter sweeps, and thus less coalescence, in the bottlenecked scenario.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}
\node[] (A) {\includegraphics[width=0.49\linewidth]{LD_s0.2_k100_u0_m0.pdf}};
\node[right = 0 of A] (B) {\includegraphics[width=0.49\linewidth]{LD_s0.2_k10_u0_m0.pdf}};
\node[below = 0 of A] (C) {\includegraphics[width=0.49\linewidth]{LD_s0.13_k100_u0_m0.pdf}};
\node[right = 0 of C] (D) {\includegraphics[width=0.49\linewidth]{LD_s0.13_k10_u0_m0.pdf}};
\node[right = 0 of A, xshift = 0.6cm, yshift = -0.5cm, draw, rectangle, rounded corners, thick, fill=gray!50, rotate = 90] (s) {\large$\bm{s=0.20}$};
\node[right = 0 of C,  xshift = 0.6cm, yshift = -0.5cm, draw, rectangle, rounded corners, thick, fill=gray!50, rotate = 90] {\large$\bm{s=0.13}$};
\node[below = -2/3 of A, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] (k) {\large$\bm{\kappa=100}$};
\node[below = -2/3 of B, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{\kappa=10}$};
\node[right = 1.9cm of k, yshift = 0.3cm, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] (R) {\fontfamily{phv}\selectfont \textbf{evol.\ rescue}};
\node[below = 0 of R, darkyellow, draw, rectangle, rounded corners, thick, fill=darkyellow!25] (Bot) {\fontfamily{phv}\selectfont \textbf{bottleneck}};
\node[below = 0 of Bot, darkred, draw, rectangle, rounded corners, thick, fill=darkred!25] {\fontfamily{phv}\selectfont \textbf{constant $\bm{N}$}};

\end{tikzpicture}
\caption{
Linkage disequilibrium, $r^2$, between neutral loci that are a recombination distance $0.001$ apart after a selective sweep from standing genetic variation in evolutionary rescue (blue; $N(0)=10^4$, $d=0.05$), in a bottlenecked population with a similar population size during the sweep (yellow; $N(0)=2945$, $d=0$), or in a population of constant size (red; $N(0)=10^4$, $d=0$).
The black line is the neutral expectation \citep[equation 7.31 in][]{wakeley2009coalescent}. 
}%
\label{fig:rescueSGV_LD}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Rescue by \textit{de novo} mutation (DNM)}
\label{sec:rescue_forward_DNM}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{The probability of rescue and soft selective sweeps}

When there are few copies of the beneficial allele at the time of environmental change rescue may depend on mutations arising \textit{de novo} at the selected site during population decline.
To predict the allele frequency and population size dynamics in this scenario we then need to derive the waiting time until the first successful rescue mutation.
Ignoring unsuccessful mutations, the first successful rescue mutation arrives according to a time-inhomogeneous Poisson process with rate, $\lambda(t) = 2 N(t) u P_\mathrm{est}$, where $2N(t) = 2N(0) e^{-d t}$ describes the decline in the number of copies of the ancestral allele.
Thus the probability that a rescue mutation has established by time $T$ (i.e., the cumulative distribution of the waiting time) is $F(T) = 1-e^{-\int_0^T \lambda(t) \mathrm{d}t}$.
Taking the limit as time goes to infinity then gives the probability of rescue \citep[c.f., equation 10 in][]{orr2008population}
\begin{equation}
P_{\mathrm{rescue}}^{\mathrm{DNM}} = 1 - e^{-2 N(0) u P_\mathrm{est} / d}.
\end{equation}

We can also calculate the probability of a soft sweep during rescue from recurrent mutation.
Taking into account the beneficial alleles present at time $t$, the rate at which additional copies arise and establish is $\lambda(t) = 2 N(t) q(t) u P_\mathrm{est}(t)$, where $2 N(t) q(t)$ is the number of ancestral alleles at time $t$ and $P_\mathrm{est}(t)$ is the probability of establishment at time $t$, which changes with allele frequency, and thus time, because allele frequency influences the genotypes the new allele experiences (i.e., its marginal fitness).
Thus the number of mutations that arise and fix is Poisson with rate $\int_0^\infty \lambda(t) \mathrm{d} t$. 
To gain intuition we make a very rough approximation, assuming $q(t)\approx1$ while mutations are arriving (i.e., while $N(t)$ is still large), so that $P_\mathrm{est}(t)\approx P_\mathrm{est}$ and we get the same Poisson rate we derived above for the first successful mutation, $\int_0^\infty \lambda(t) \mathrm{d} t \approx 2N(0) u P_\mathrm{est}/d$.
The resulting probability distribution for the number of mutations that establish is analogous to the result in a model with haploid selection \citep[c.f., equation 7 in][]{wilson2017soft}.  
Dividing the expected number of establishing mutations by the probability of rescue, the expected number that establish given rescue is
\begin{equation}\label{eq:ENestDNM}
n^\mathrm{DNM}_\mathrm{rescue} = -\frac{\log(1-P_\mathrm{rescue}^\mathrm{DNM})}{P_\mathrm{rescue}^\mathrm{DNM}}
\end{equation}
Ignoring density-dependence, our approximation will underestimate the number of establishing mutations when $h\neq0$ since selection in heterozygotes will prolong the persistence of the ancestral allele (creating more opportunity for mutation) and establishment probabilities will rise with the frequency of the beneficial allele (because more mutant homozygotes are then created).
At the same time, if the carrying capacity is reached before fixation, our simple form of density-dependence will introduce additional genetic drift and hence tend to reduce the number of mutations that establish.
In the end we find our rough approximation to underestimate the number of establishing mutations (Figure \ref{fig:PsoftDNM}B), suggesting that our underestimate of the probability of establishment has a larger effect than the excess drift brought about by the carrying capacity for these parameter values.

With these same approximations the probability of a soft selective sweep given rescue (i.e., the probability more than one copy establishes) is 
\begin{equation}\label{eq:PsoftDNM}
P_{\mathrm{soft}|\mathrm{rescue}}^{\mathrm{DNM}} = 1 - n^\mathrm{DNM}_\mathrm{rescue} (1-P_{\mathrm{rescue}}^{\mathrm{DNM}}),
\end{equation}
as in a haploid population \citep[equation 8 in][]{wilson2017soft}.
This approximation is, however, an underestimate in diploid populations (Figure \ref{fig:PsoftDNM}A), where the creation of homozygote mutants can inflate establishment probabilities.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}
\node[] (A) {\includegraphics[width=0.49\linewidth]{PsoftDNM.pdf}};
\node[below = -1cm of A] (B) {\includegraphics[width=0.49\linewidth]{NumberEstDNM.pdf}};
\node[below = -1cm of A, xshift = -0.75cm, darkred, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont\textbf{constant $\bm{N}$}};
\node[right = 0cm of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont\textbf{evol.\ rescue}};
\draw[very thick] (-1,-2.5) -- (-0.25,-2.5); 
\draw[fill=black] (0,-2.5) circle (0.05cm);
\node at (0.75, -2.475) {$s=0.2$}; 
\draw[very thick] (-1,-2.9) -- (-0.75,-2.9); \draw[very thick] (-0.5,-2.9) -- (-0.25,-2.9); 
\draw[very thick] (0,-2.9) circle (0.05cm);
\node at (0.84, -2.875) {$s=0.13$}; 
\end{tikzpicture}
\caption{
Soft sweeps from \textit{de novo} mutation.
(A) The probability that more than mutant lineage has descendants at the time of fixation in rescue (blue; $d=0.05$) or in a population of roughly constant size (red; $d=0$) as a function of the mutation rate, $u$.
(B) The expected number of mutant lineages that have descendants at the time of fixation.
The blue curves are Equations \ref{eq:PsoftDNM} (panel A) and \ref{eq:ENestDNM} (panel B).
The red curves are from Ewens' sampling formula \citep[equation 11 (panel A) and equation 12 (panel B) in][with $\theta = 2N_e(0) u = 8 N(0) u/7$ and $n=2N(0)$]{pennings2006soft2}.
Each point is based on 100 replicate simulations where a sweep (and population recovery) was observed.
Error bars in (B) are standard errors.
Parameters: $N(0)=10^4$.
}%
\label{fig:PsoftDNM}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Effective initial allele frequency and the backward-time dynamics}

Following \cite{orr2014population} (see their Text S2 and our File S1 for more detail), taking the derivative of the cumulative distribution of waiting times for the first successful mutation, $F(T)$, and dividing by the probability of rescue gives the probability distribution function for the arrival time of the first establishing rescue mutation given rescue, $f(t)$.
As discussed above, while the first establishing mutation is still rare it will grow exponentially at rate $\epsilon = (1+s h)(1-d) - 1$, and conditioned on its establishment will very quickly reach $1/P_\mathrm{est}$ copies. 
Integrating over the distribution of arrival times then gives the expected number of copies of this successful mutation at time $t$ since the environmental change given rescue, $\int_0^\infty (e^{(t-\tau) \epsilon} / P_\mathrm{est}) f(\tau) \mathrm{d}\tau$.
Solving this integral, evaluating at $t=0$,and dividing by the total number of alleles at the time of environmental change, $2N(0)$, it is therefore as if the successful mutation was present at the time of environmental change, with frequency
\begin{equation}\label{eq:p0dnm}
p_{0|\mathrm{rescue}}^{\mathrm{DNM}} = \frac{1}{2N(0)}\frac{1}{P_\mathrm{est}} \frac{(1-P_{\mathrm{rescue}}^{\mathrm{DNM}}) \left[\Gamma\left(1+\epsilon/d,\log(1-P_{\mathrm{rescue}}^{\mathrm{DNM}})\right) - \Gamma\left(1+\epsilon/d\right) \right]}{P_{\mathrm{rescue}}^{\mathrm{DNM}} \log(1-P_{\mathrm{rescue}}^{\mathrm{DNM}})^{\epsilon/d}},
\end{equation}
where $\Gamma(z)$ is the gamma function \citep[equation 6.1.1 in][]{abramowitz1972handbook} and $\Gamma(a,x)$ is the incomplete gamma function \citep[equation 6.5.3 in][]{abramowitz1972handbook}.
The factor $1/P_\mathrm{est}>1$ increases the effective initial frequency, because we have conditioned on establishment, while the last factor decreases the effective initial frequency, because we must wait for the mutation to arise.
When the expected number of rescue mutations, $-\log(1-P_{\mathrm{rescue}}^{\mathrm{DNM}}) = 2 u N(0) P_\mathrm{est} / d$, is small this expected number cancels out and the last factor in Equation \ref{eq:p0dnm} becomes approximately $d/(d+\epsilon)$, which is independent of the mutation rate \citep[as in the haploid case;][]{orr2014population}.
That is, conditioning on unlikely rescue, rescue mutations arise earlier in populations that decline faster.

In a population of constant size, $N=N(0)$, the first successful mutation arrives at a time-homogeneous rate, $\lambda=2 N(0) u P_\mathrm{est}$, such that the probability distribution of its waiting time is a simple exponential, $f(t) = \lambda e^{-\lambda t}$.
Integrating the expected number of copies of the allele over the distribution of waiting times shows that the waiting time factor in a population of roughly constant size is $4N(0)u/(1+4N(0)u)$, which, in contrast to rescue, depends strongly on mutational input but is independent of establishment probability.
However, our goal here is to determine the genetic signatures of sweeps, which require only a prediction of the backward-time dynamics from the point of fixation. 
In the case of constant population size, the time of establishment of the first successful mutation is therefore irrelevant for genetic signatures (in contrast, when $d>0$ we need to predict the time of establishment to predict the population size dynamics; Equation \ref{eq:backwardsptnt}). 
For populations of constant size we therefore need only determine the effective allele frequency at the time the sweep begins, which is simply $1/(2N(0)P_\mathrm{est})$, independent of the mutation rate.

Figure \ref{fig:rescueDNM_dynamics} compares our analytical approximations (Equations \ref{eq:backwardsptnt} and \ref{eq:p0dnm}) against individual-based simulations.
As with rescue from standing genetic variance (Figure \ref{fig:rescueSGV_dynamics}), the non-linearities of the deterministic phase cause our semi-deterministic approximation to underestimate mean allele frequencies and overestimate mean population sizes as the allele frequency declines from fixation.
We again see that rescue sweeps tend to be shorter than those in populations of constant size (due to lower establishment probabilities) and that the bottleneck sizes and sweep times given rescue depend little on the probability of rescue.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{The structured coalescent}

Figure \ref{fig:rescueDNM_timing} shows the timing of coalescence, recombination, and mutation for a sample of size 2 at a linked neutral locus. 
The patterns of coalescence and recombination are essentially the same as observed during sweeps from standing genetic variation (Figure \ref{fig:rescueSGV_timing}).
The timing of mutation is qualitatively similar to that of coalescence, peaking near establishment (since both depend on $1/p$).
The main effect of the excess coalescence during rescue is a reduced probability of mutating off the sweep (compare areas under dotted curves), which is exacerbated by rescue sweeps being shorter.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Genetic signatures at linked neutral loci}

Figure \ref{fig:rescueDNM_heterozygosity_relative} shows relative pairwise nucleotide diversity around the selected site.
The patterns here are nearly identical to those found after sweeps from standing genetic variation (Figure \ref{fig:rescueSGV_heterozygosity_relative}), although the difference between rescue and the constant population size model are larger here, perhaps partly due to the reduction in rates of mutation off the selected sweep (and also because our parameter choice causes the bottlenecks to be deeper when rescue occurs by mutation).
Figure \ref{fig:rescueDNM_tajimasD} shows Tajima's $D$ values around the selected site, which are nearly identical to those observed after sweeps from standing genetic variation (Figure \ref{fig:rescueSGV_tajimasD}).

On a related note, \cite{wilson2017soft} reasoned that, because soft sweeps from recurrent mutation are expected when rescue is likely while hard sweeps are expected when rescue is rare (Equation \ref{eq:PsoftDNM}), population bottlenecks will tend to be more extreme when rescue occurs by a hard selective sweep.
From this they argued it might actually be easier to detect soft sweeps from patterns at linked neutral loci, as bottlenecks are expected to obscure the signal.
Here we show the importance of conditioning on rescue, which roughly equalizes bottleneck sizes across scenarios with very different probabilities of rescue (e.g., due to differences in mutation rate; Figure \ref{fig:rescueDNM_dynamics}), potentially making harder sweeps easier to detect due to their greater effect on local gene genealogies (Figures \ref{fig:rescueDNM_timing}-\ref{fig:rescueDNM_tajimasD}).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Rescue by migrant alleles (MIG)}
\label{sec:rescue_forward_mig_main}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{The probability of rescue and soft selective sweeps}

Rescue can also arise from beneficial alleles that arrive via migration.
Assuming that the number of migrant alleles that replace a resident allele each generation is Poisson with mean $m$, the waiting time until the first successful migrant is exponential with rate $\lambda = m P_\mathrm{est}$.
Given the population is expected to persist in the absence of beneficial alleles for $\log(N(0))/d$ generations, the probability of rescue is therefore roughly the probability the first successful migrant arrives by then,
\begin{equation}
P_\mathrm{rescue}^\mathrm{MIG} = 1 - N(0)^{-m P_\mathrm{est}/d}.
\end{equation}

Under rescue from standing variation or new mutation we derived the probability and extent of soft selective sweeps from the forward-time process. 
In contrast, under rescue from migration we use the coalescent.
In particular, Equation \ref{eq:coal_rates} shows that the per generation probability of migration and coalescence depend on population size and beneficial allele frequency in the same way. 
This similar form implies that the relative rates at which lineages coalesce and migrate at the selected site does not depend on the population size and allele frequency. 
\cite{pennings2006soft2} used this fact to show that, in an ideal population of constant size, the number of unique migrant haplotypes contributing to a present day sample, as well as their proportions, is described by Ewens' sampling formula \citep[pages 334\textit{ff} in][]{ewens2004mathematical} when we replace $\theta$ with $2 m$. 
Powerfully, this results holds even in non-ideal populations of changing size \citep[as briefly noted by][p.\ 1081-1082]{pennings2006soft2} -- including during evolutionary rescue -- as long as the relationship between the effective and census population sizes remains the same (i.e., if $N_e(t)/N(t)$ is a constant; in which case we now replace $\theta$ with $2m N_e/N$ in Ewens' sampling formula). 
Thus the softness of a sweep from migration depends only on the migration rate and variance in gamete numbers ($\sigma^2$, which determines $N_e/N$; see \nameref{sec:pestNe}), and is the same during rescue as it is in a population of constant size (i.e., it is independent of $d$).
The analogous result for rescue by \textit{de novo} mutation does not hold \citep[as it does for a population of constant size,][]{pennings2006soft2}, since the rate of mutation is not inversely proportional to population size (Equation \ref{eq:coal_rates}).  

Here we use just two properties of Ewens' sampling formula, the expected number of unique migrants among a sample of size $n$ \citep[page 336 in][]{ewens2004mathematical},
\begin{equation}\label{eq:ewens_number}
n^\mathrm{MIG}_\mathrm{rescue} = \sum_{j=1}^n \frac{2 m N_e/N}{j - 1 + 2 m N_e/N},
\end{equation}
and the probability that this is more than two \citep[equation 10.9 in][]{ewens2004mathematical},
\begin{equation}\label{eq:ewens_Psoft}
P_{\mathrm{soft}|\mathrm{rescue}}^\mathrm{MIG} = 1 - \prod_{j=1}^{n-1} \frac{j}{j + 2 m N_e/N}.
\end{equation}
Figure \ref{fig:PsoftMIG} shows that these formulas perform very well, even when we sample the entire population ($n=2N(0)$).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}
\node[] (A) {\includegraphics[width=0.49\linewidth]{PsoftMIG.pdf}};
\node[below = -1cm of A] (B) {\includegraphics[width=0.49\linewidth]{NumberEstMIG.pdf}};
\node[below = -1cm of A, xshift = -0.75cm, darkred, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont\textbf{constant $\bm{N}$}};
\node[right = 0cm of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont\textbf{evol.\ rescue}};
\draw[fill=black] (0,-2.5) circle (0.05cm);
\node at (0.75, -2.475) {$s=0.2$}; 
\draw[very thick] (0,-2.9) circle (0.05cm);
\node at (0.84, -2.875) {$s=0.13$}; 
\end{tikzpicture}
\caption{
Soft sweeps from migrant alleles.
(A) The probability that more than migrant lineage has descendants at the time of fixation in rescue (blue; $d=0.05$) or in a population of roughly constant size (red; $d=0$) as a function of the migration rate, $m$.
(B) The expected number of migrant lineages that have descendants at the time of fixation.
The curves are Equations \ref{eq:ewens_Psoft} (panel A) and \ref{eq:ewens_number} (panel B) using $n=2N(0)$ (i.e., sampling the entire population).
Each point is based on 100 replicate simulations where a sweep (and population recovery) was observed.
Error bars in (B) are standard errors.
Parameters: $N(0)=10^4$.
}%
\label{fig:PsoftMIG}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Effective initial allele frequency and the backward-time dynamics}

Dividing the (truncated) exponential waiting time distribution for the first successful migrant allele by the probability of rescue gives the waiting time distribution conditioned on rescue.
Following the same approach as above (Section \nameref{sec:rescue_forward_DNM}), the effective initial frequency of the beneficial allele given rescue is then
\begin{equation}\label{eq:p0m}
p_0^\mathrm{MIG} = \frac{1}{2N(0)}\frac{1}{P_\mathrm{est}}\frac{2m \left( 1 - (1-P_\mathrm{rescue}^\mathrm{MIG})  N(0)^{-P_\mathrm{est}/(2d)} \right)}{(1+2m)P_\mathrm{rescue}^\mathrm{MIG}}.
\end{equation}
When the migration rate is small this last factor is nearly independent of $m$ (analogous to the mutation case).
In a population of constant size the waiting time to the first successful migrant allele is simply exponential with rate $\lambda = m P_\mathrm{est}$, giving a waiting time factor $2m/(1+2m)$, which is strongly dependent on $m$ but independent of $P_\mathrm{est}$.
However, as in the case of \textit{de novo} mutation, to characterize the backward-time dynamics we do not need to know when the sweep begins in a population of constant size, just the effective initial frequency at this time, $1/(2N(0)P_\mathrm{est})$.

Figure \ref{fig:rescueMIG_dynamics} compares our analytical approximations (Equations \ref{eq:backwardsptnt} and \ref{eq:p0m}) against individual-based simulations.
We see the predictions do fairly well for larger values of $m$, but can fair quite poorly with small $m$.
In the latter case the first successful migrant allele sometimes arrives when the population is so small that the beneficial allele increases in frequency much faster than the deterministic expectation.
Here we enter a different regime, which we do not attempt to approximate.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{The structured coalescent}

Figure \ref{fig:rescueMIG_timing} shows the timing of migration relative to recombination and coalescence (Equation \ref{eq:Pikt}).
As with rescue from standing genetic variance or mutation (Figures \ref{fig:rescueSGV_timing}-\ref{fig:rescueDNM_timing}), the bottleneck increases the overall coalescence rate and shifts its timing closer to fixation, overlapping more with recombination. 
Migration scales with coalescence (Equation \ref{eq:coal_rates}) and is thus similarly increased and shifted.
When migration rates are small enough that the migrant allele tends to fix when the population is very small we observe an initial spike in coalescence, which is expected to greatly reduce diversity.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsubsection*{Genetic signatures at linked neutral loci}

The genetic signatures at linked neutral loci produced by a sweep that arises from migration depends on the history of the metapopulation, e.g., how and when the sweeps occurred in each patch and the historical migration rates between them.
We therefore omit an exploration of the signatures produced in this scenario, which deserves a full and careful treatment on its own.
Previous work has explored some potential signatures of migrant sweeps in populations of constant size.
For example, if the historic migration rate between the migrant and focal population is low then we should expect few migrant alleles away from the selected site.
Then, under low contemporary migration rates we expect a relatively hard sweep and the so-called ``volcano" pattern of genetic diversity \citep{setter2019volcanofinder}, where diversity is maximized at an intermediate distance from the selected site due to a more balanced presence of both migrant and non-migrant alleles.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Discussion}
\label{sec:discussion}

%%major findings
Here we have explored genetic signatures of evolutionary rescue by a selective sweep. 
By allowing demography to depend on the absolute fitness of the genotypes that comprise the population we explicitly invoke a feedback between demography and evolution.
This feedback restricts the range of dynamics, and thus the signatures, that one should expect to observe.
We find that, because the probability of establishment for an allele with a given selective advantage is reduced in declining populations \citep[Equation \ref{eq:pest}; see also][]{otto1997probability}, selective sweeps causing rescue are expected to be harder than those in populations of constant size when sweeps arise from standing genetic variance or recurrent mutation (Figures \ref{fig:PsoftSGV} and \ref{fig:PsoftDNM}; consistent with \citealp{wilson2014soft} and \citealp{wilson2017soft}).
Further from the selected locus, the demographic bottleneck experienced during rescue increases rates of coalescence relative to mutation and recombination (Figures \ref{fig:rescueSGV_timing} and \ref{fig:rescueDNM_timing}), creating wider dips in relative diversity (Figures \ref{fig:rescueSGV_heterozygosity_relative} and \ref{fig:rescueDNM_heterozygosity_relative}) and lower absolute diversity genome-wide (Figure \ref{fig:rescueSGV_heterozygosity_empirical}; consistent with \citealp{innan2004pattern}). 
Like absolute diversity, Tajima's $D$ captures both the hardening of the sweep and the demographic bottleneck, causing $D$ to generally reach both higher and lower values under rescue (Figures \ref{fig:rescueSGV_tajimasD} and \ref{fig:rescueDNM_tajimasD}).
These differences between evolutionary rescue and standard sweeps all become larger under weaker selection (i.e., when the heterozygote has a smaller growth rate, $s/2-d\ll1$) as the slower sweeps that result imply deeper, longer bottlenecks during rescue.
If we instead compare rescue to populations bottlenecked to the same effective population size during the sweep, we find that the remaining signatures of rescue are harder sweeps that take longer to complete, allowing more time for coalescence and hence lower genome-wide diversity and elevated linkage disequilibrium (Figure \ref{fig:rescueSGV_LD}).
The subtle difference in the timing of recombination between the two scenarios, with recombination overlapping more with coalescence during rescue, may also increase variation in the number of derived mutations at a site \citep[as expected following soft sweeps from recurrent mutation in populations of constant size;][]{pennings2006soft}, giving rise to site frequency spectra with slightly ``flatter bottoms" (Figure \ref{fig:rescueSGV_SFS}).

In contrast to standing variance or mutation, when sweeps arise from a constant rate of migration demography has no affect on the number of beneficial alleles that establish \citep[as briefly noted by][]{pennings2006soft2} and thus rescue has no affect on the hardness of the sweep (Figure \ref{fig:PsoftMIG}). 
Further, because the rates of coalescence and migration are both inversely proportional to the number of beneficial alleles ($N(t) p(t)$, Equation \ref{eq:coal_rates}, Figure \ref{fig:rescueMIG_timing}), the distribution of the number and frequency of migrant haplotypes spanning the selected site is given by Ewens' sampling formula \citep{ewens1972sampling}, with mutation replaced by migration \citep{pennings2006soft2}.
The signatures of rescue by migration at linked neutral loci depend on the demographic and selective history of the metapopulation, which we do not explore here.

%%unanswered questions
Evolutionary rescue has been explored theoretically \citep[e.g.,][]{gomulkiewicz1995does,uecker2016role,anciaux2018evolutionary} and observed repeatedly in both experiments \citep[e.g.,][]{bell2009evolutionary,lindsey2013evolutionary,ramsayer2013evolutionary} and in host-pathogen systems in nature \citep[e.,g.,][]{wei1995viral,feder2016more}.
More recently, a number of studies have examined genetic data following purported evolutionary rescue in the wild, including bats altering hibernation to survive white-nose syndrome \citep{gignoux2018genomic}, killifish deleting receptors to tolerate pollution \citep{oziolor2019adaptive}, and tall waterhemp evolving herbicide resistance \citep{kreiner2019multiple}.
In the latter two cases there is strong evidence of a recent selective sweep by a very beneficial allele \citep[in one of these cases the evidence includes reduced nucleotide diversity at the selected site;][]{kreiner2019multiple}.
Genetic evidence for a putatively recent demographic bottleneck was presented in only one of these studies \citep{oziolor2019adaptive}, being detected from genome-wide reductions in nucleotide diversity and increases in Tajima's $D$ relative to populations that did not experience the same environmental change.
Developing a statistical model to fit our theory to such data in a more quantitative way could lead to parameter estimates, which could then inform how likely rescue may have been (or if persistence was assured regardless, $d\approx0$).
To do so one would also have to incorporate the time since fixation as the site frequency spectra and linkage disequilibrium will begin a return to their neutral expectations following the sweep and bottleneck \citep[e.g., the sweep signatures in Tajima's $D$ and linkage disequilibrium decay within $\sim2N_e$ generations;][]{przeworski2002signature}.
Ignoring the time since fixation would cause underestimates of $s$ and $d$ and make rescue look more like a sweep in a constant population.

A lingering question that helped motivate this study is whether one could ever infer evolutionary rescue from genetic data alone.
Such a possibility would greatly help assess the relevance of rescue in nature.
In the empirical examples just discussed we implied that evidence for a sweep and a bottleneck is consistent with rescue.
Yet stronger support for rescue would come from inferring the coincident timing of the sweep and bottleneck.
Our analysis suggests that detecting coincident sweeps and bottlenecks from site frequency spectra and linkage disequilibrium will be difficult as the relative timing of coalescence and recombination leaves only subtle signatures.
Consistent with this, explicit estimates of the timing of a sweep \citep[e.g.,][]{ormond2016inferring} or a bottleneck \citep[reviewed in][]{beichman2018using} given a genetic sample collected from a single time point come with relatively large amounts of uncertainty.
Sampling before and after the potential rescue event may provide one way forward, as it would help tighten the bounds on the timing of the bottleneck and sweep.
However, it should be noted that even if the sweep and bottleneck appear to have co-occurred, this correlation in timing does not imply it was caused by a feedback between demography and evolution. 
It is, of course, difficult to say in any case -- without observing replicate populations go extinct or performing experiments -- whether extinction would have occurred (or will occur) without adaptive evolution, as required by the strict definition of evolutionary rescue.
More experiments \citep[such as][]{rego2019dynamics} that explore the genetic consequences of verified rescue may help develop a robust genetic signal of the feedback between demography and evolution during rescue, or at least determine if one can be detected in real genomes.

%%conflicts
A strength of the above analysis is that we have explicitly modelled a feedback between demography and evolution, restricting the range of genetic signatures we consequently expect to observe. 
To take a recent example, \cite{harris2018unfounded} have claimed that the lower reductions in genetic diversity within HIV populations adapting to less efficient drugs \citep[as observed by][]{feder2016more} could be due to weaker bottlenecks or slower sweeps rather than sweeps being softer, i.e., arising from multiple mutations. Fortunately, in this case genetic time-series data were available to show that the ability of HIV to reliably adapt on a short time-scale necessitates mutation rates and selection coefficients that imply adaptation by soft sweeps is likely \citep{feder2018clarifying}. 
Formally modeling a feedback between demography and evolution also helps narrow the relevant parameter range.
For example, under a haploid version of the model explored here (as is applicable to HIV) the minimum population size during rescue by new mutations is $N(0) s (2N(0)s)^{-d/s} / (s-d)$ \citep[equation 22 in][]{orr2014population}.
Thus, for a given initial population size, $N(0)$, and selection coefficient, $s$, the minimum expected population size is $e \ln S / (2s)$, where $S=2N(0)s$, implying that the minimum population size consistent with the model is roughly proportional to $1/s$.
The imposed feedback between demography and evolution therefore precludes simultaneously slow sweeps and large bottlenecks (for example negating the two smallest bottleneck sizes in figure 3A of \citealp{harris2018unfounded} and constraining one to the upper right portion of figure 3A-B in \citealp{feder2018clarifying}).
While it is very likely in this case that soft sweeps are indeed the cause of the pattern \citep{feder2018clarifying}, incorporating an explicit model of how demography and evolution interact could help focus future debates.

%%uncertainty, ignorance, confusion
The model presented here is but one model of evolutionary rescue, which involved a number of important assumptions.
One of these is that the beneficial allele acts multiplicatively with the ancestral background, so that the absolute fitness its carriers, $(1+s h)(1-d)$ and $(1+s)(1-d)$, -- and thus the probability it establishes -- are affected by the decline rate of the ancestral genotype, $d$.
Meanwhile, this same assumption caused the relative fitness of the mutants, $1+sh$ and $1+s$, -- and thus the allele frequency dynamics during the sweep, once started -- to be independent of the initial decline rate (Equation \ref{eq:deltapdeltaN}).
If, instead, we made the absolute fitness of the heterozygote and mutant homozygote independent of the initial decline rate, say $1 + s h$ and $1 + s$, then the reverse would be true; the relative fitness of the mutant would depend on the initial rate of population decline, $d$, while its absolute fitness while establishing would not. 
We expect these effects would, however, largely cancel out (higher establishment probabilities will cause longer sweeps and more coalescence, but slower changes in allele frequency will allow more recombination off the sweep).
In any case, because our results depend primarily on the absolute and relative fitness of the heterozygote, the alternative model just described may closely match the model analyzed in detail here when $s h$ is replaced by $(s h+d)/(1-d)$.

A second key assumption we have made is that the beneficial allele acts additively with the ancestral allele at that locus ($h=1/2$).
This assumption has been made for convenience; it yields a simple explicit approximation of the allele frequency and population dynamics (Equation \ref{eq:backwardsptnt}) that allows closed-form solutions of the structured coalescent (Equation \ref{eq:Pikt}), greatly accelerating our computation of the coalescent and pairwise diversity (Equation \ref{eq:EH_rescue}).
Alternative forms of dominance are, however, likely and will impact our results.
At one extreme, a completely recessive beneficial allele ($h=0$) is much less likely to establish when compared to additivity, even in a constant population \citep[compare Equation \ref{eq:pest} with $v=1$ and $\epsilon=s/2$ to equation 15 in][]{kimura1962probability}, making rescue nearly impossible in outcrossing populations \citep{uecker2017evolutionary}.
In fact, any $h < d/s$ will cause the heterozygote to have a negative growth rate (i.e., be subcritical), meaning that establishment will rely on the stochastic persistence of subcritical lineages until they create a supercritical mutant homozygote, analogous to the fixation of underdominant alleles or chromosomal rearrangements.
As the establishment of such alleles in time to rescue the population is extremely unlikely (compared to supercritical heterozygotes), we have neglected this parameter range here in order to focus on parameter values that are more likely to be reflected in empirical systems where rescue has occurred.
It may, however, be possible to combine our approach here with the approach taken for sweeps with arbitrary dominance in populations of constant size \citep{ewing2011selective} and the probability of establishment of underdominant alleles \citep[e.g., equation 3 in][]{lande1979effective}.
At the other extreme, complete dominance ($h=1$) will greatly increase the probability of establishment and rescue \citep{uecker2017evolutionary}, as well as population mean fitness and thus population size.
All else equal, we therefore expect rescue to have less effect on the signatures of selective sweeps relative to those in populations of constant size when the rescuing allele is more dominant.
Given that the marginal fitness of the beneficial allele will not depend on allele frequency under complete dominance, we expect dynamics much like the haploid model \citep{orr2014population}, where simple predictions of allele frequency and population size are more accurate.

Finally, it is of course possible to model rescue under much more complex lifecycles and population structure \citep[e.g., as expected for the evolution of malarial drug resistance;][]{kim2014population}, at least using simulations. 
More complex lifecycles, such as those of parasites like \textit{Plasmodium} and HIV, could cause the bottleneck to have additional impacts on the resulting genetic signature. 
For example, our populations are obligate sexual out-crossers, such that the probability of recombination does not depend on the population size (c.f., Equation \ref{eq:coal_rates}), as everyone must mate with some one. 
However, with selfing and/or facultative sex (genetic exchange), rates of recombination could be lower at lower population densities (as expected for HIV), which would increase the impact of bottlenecks on resulting genetic signatures.

%extensions
Evolutionary rescue is only one example of a myriad of processes where demography and evolution feedback on one another. 
Our approach -- combining forward-time eco-evolutionary models with coalescent theory to predict genetic signatures -- could be used in many other scenarios.
For instance, adaptive colonization of new habitat (a.k.a., adaptive niche expansion) is a closely related process for which a similar approach has already been taken \citep{kim2010signatures}.
As in the case of rescue, explicitly modelling the feedback between demography and evolution in adaptive niche expansion changes the expected signatures left behind by selective sweeps as compared to (bottlenecked) Wright-Fisher populations.
Such an approach is interesting from a conceptual point-of-view, improving our understanding of how eco-evolutionary dynamics affect genetic signatures.
But further, given the computational power and simulation platforms available today, it is no longer necessary to restrict oneself to Wright-Fisher populations; researchers may now simulate under much more ecologically-realistic models.
Developing simple approximations in parallel will become increasingly important for intuition as our models become more complex.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Acknowledgements}
We thank Pleuni Pennings and Alison Feder for helpful discussions.
We thank Ben Haller for comments on our SLiM code.
Financial support was provided by the Center for Population Biology at the University of California - Davis (fellowship to MMO), Banting (Canada; fellowship to MMO), and the National Institute of General Medical Sciences of the National Institutes of Health (NIH R01 GM108779, awarded to GC).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%\newpage
\bibliography{biblio}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\setcounter{figure}{0}
\renewcommand{\thefigure}{A\arabic{figure}}
\setcounter{table}{0}
\renewcommand{\thetable}{A\arabic{table}}
\setcounter{section}{0}
\renewcommand{\thesection}{A\arabic{section}}
\setcounter{equation}{0}
\renewcommand{\theequation}{A\arabic{equation}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Supplementary text: Simulated lifecycle}
\label{sec:lifecycle}

Here we describe the lifecycle used in our simulations, which mimics that used in previous models of evolutionary rescue in gradually changing environments \citep{burger1995evolution}.
Let $n_i(t)$ be the number of individuals with genotype $i\in\{aa,Aa,AA\}$ at the beginning of generation $t$, with $N(t) =  \sum_i{n_i(t)}$ the total population size.
Viability selection, where genotype $i$ survives with probability $V_i\in[0,1]$, is assumed to occur before reproduction.
Each surviving individual then ``mothers" $B$ offspring, each with a randomly chosen mate (possibly oneself), and each mating produces a single offspring. 
If there are more than $N(0)$ offspring (we assume the population starts at carrying capacity) we randomly choose $N(0)$ to begin the next generation.

We next describe the deterministic allele frequency dynamics.
Letting $p_{j,k}(i)$ be the probability a mating between genotypes $j$ and $k$ produces an offspring with genotype $i$, the expected number of individuals of genotype $i$ after reproduction is then  
\begin{equation}
n_i^*(t) = \sum_{j,k} \tilde{n}_j(t) B \frac{\tilde{n}_k(t)}{\tilde{N}(t)}p_{j,k}(i),
\end{equation}
where $\tilde{n}_i(t) = n_i(t) V_i$ is the expected number of individuals with genotype $i$ and $\tilde{N}(t) = \sum_i{\tilde{n}_i(t)}$ is the expected population size after viability selection.
Assuming fair Mendelian transmission, the expected number of $A$ alleles after reproduction is $n_{Aa}^*(t) + 2n_{AA}^*(t) = W_{Aa} n_{Aa}(t) + 2 W_{AA} n_{AA}(t)$, using $W_i = V_i B$, which we refer to as the fitness of genotype $i$.
Given that the total number of alleles after reproduction is expected to be $2N^*(t) = 2\sum_i n_i(t) W_i = 2N(t) \overline{W}(t)$, where $\overline{W}(t)$ is the mean population fitness at the beginning of generation $t$, the expected frequency of allele $A$ after reproduction is
\begin{equation}\label{eq:pnext_rescue}
\begin{aligned}
p^*(t) &= \frac{\frac{1}{2} W_{Aa} p_{Aa}(t) + W_{AA} p_{AA}(t)}{\overline{W}(t)},
\end{aligned}
\end{equation}
where $p_i(t)=n_i(t)/N(t)$ is the frequency of genotype $i$ in generation $t$.
Since density-dependence is random it does not change this expectation, so that the allele frequency in next generation is $p(t+1)=p^*(t)$.
Thus the allele frequency dynamics are the same as those in a population of constant size with relative fitnesses $W_i$ \citep[equation 5.2.3 in][]{crow1970introduction}.
Further, one can use Equation \ref{eq:pnext_rescue} to show that the genotype frequencies are expected to remain in Hardy-Weinberg proportions, allowing us to capture the dynamics of the whole system by tracking only the expected changes in the frequency of allele $A$ and total population size (Equation \ref{eq:deltapdeltaN}).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Supplementary text: Genetic drift in the simulated lifecycle}
\label{sec:pestNe}

In our simulated lifecycle, in generation $t$ a rare allele with a viability of $V_i$ survives to reproduction with probability $V_i$, and given so, is present in $Y+Z$ offspring, where $Y$ is binomial with $B$ trials and probability of success $1/2$ (number of offspring mothered and Mendelian segregation), and $Z$ is binomial with parameters $B N(t)$ and $(1/N(t))/2$ (randomly chosen as a father and Mendelian segregation).
With $\overline{V}$ the population mean viability, the expected number of offspring produced is $N(t) B \overline{V} = N(t) \overline{W}$.
When this is less than $N(0)$ there is no density dependence, so that all offspring survive to the next generation, while if $N(t) \overline{W} > N(0)$ we randomly choose $N(0)$ offspring to begin the next generation, implying each survives with probability $N(0)/(N(t)\overline{W})$.

From this we can calculate the mean and variance in the number of mutant alleles contributed to the next generation by a rare mutant allele in an individual with viability $V_i$.
In the absence of density dependence (e.g., in a declining population, where $\overline{W} < 1$) the expected number of copies of the allele contributed to the next generation is $W_i = B V_i$ and the variance is $W_i(3+4(B-W_i))/4 + O(1/N(t))$.
In a large population the heterozygote therefore has growth rate $\epsilon = W_{Aa} - 1$ and variance $v = W_{Aa}(3+4(B-W_{Aa}))/4$, which we use in calculating its probability of establishment (Equation \ref{eq:pest}).

With density dependence (i.e., when $N(t)\overline{W}>N(0)$) the expected number of copies contributed to the next generation is $(W_i / \overline{W})(N(t) / N(0))$ and the variance is $(W_i(3+4(B-W_i))/4)(N(t)/(N(0)\overline{W}))^2 + O(1/N(t))$.
When the current population size is $N(0)$ these reduce to $W_i / \overline{W}$ and $W_i(3+4(B-W_i))/(4\overline{W}^2) + O(1/N(t))$.
Thus the backward-time growth rate of the heterozygote in a large population of mutant homozygotes at carrying capacity is $\epsilon = 1 - W_{Aa}/W_{AA}$ and its variance is $W_{Aa}(3+4(B-W_{Aa}))/(4W_{AA}^2)$, which we use in calculating the probability of establishment (Equation \ref{eq:pest}) for the effective final allele frequency (Equation \ref{eq:pf}).

We can also use the variance in the number of gametes contributed to the next generation to calculate the effective population size.
In a large population with weak selection ($W_i\approx\overline{W}\approx1$) the variance is $\approx (3+4(B-1))/4$ regardless of whether the population is at carrying capacity.
We therefore use $\sigma^2 = (3+4(B-1))/4$ to calculate the effective population size, $N_e(t)/N(t) \approx 4/(2+\sigma^2)$ (Section \nameref{sec:rescue_backward}).
Throughout we use $B=2$, meaning that $\sigma^2 \approx 7/4$, and thus our model imparts nearly twice as much drift as a Wright-Fisher population (where $\sigma^2\approx1$).

Note that drift increases with $B$ since larger $B$ imply that fewer individuals survive viability selection ($V= W/B$) but those that do have more offspring.
To keep our model close to a Wright-Fisher population we use the smallest value of $B$ consistent with long-term population persistence, $B>1$.
An alternative model where the expected number of offspring upon reproduction is any positive real number, say Poisson with mean $B$, would allow rates of drift closer to that of a Wright-Fisher population (by setting $B$ closer to 1).

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Supplementary text: Simulation details}
\label{sec:simdetails}

Forward-time simulations of the life-cycle described in \nameref{sec:lifecycle} were performed in \texttt{SLiM} \citep[version 3.3;][]{Haller2019slim} with tree-sequence recording \citep{Haller2019tree}.

We simulated 20 Mb chromosomes with the selected locus one of the centre bases, all other sites were neutral.
We assumed a per base pair recombination rate of $r_\mathrm{bp}=2\times10^{-8}$ \citep[i.e., 2 cM/Mb; e.g.,][]{mackay2012drosophila} and per base mutation rate at neutral loci of $U=6\times10^{-9}$ \citep[e.g.,][]{haag2007direct}.
The recombination rate between two loci $n$ bases apart was calculated as the probability of an odd number of crossover events assuming $n$ independent Bernoulli trials, $r = (1-(1-2r_\mathrm{bp})^n)/2 \approx (1-e^{2 r_\mathrm{bp} n})/2$ \citep[equation 3 in][]{haldane1919combination}, i.e., no crossover interference.

A population was considered rescued (or a sweep complete) when the beneficial mutation was fixed and the population size had recovered to $N(0)$.
Once a population was rescued we used \texttt{msprime} \citep{kelleher2016efficient} to recapitate the population (simulate the neutral coalescent back in time from the start of the forward-time simulation, until all sites had fully coalesced) using an effective population size of $N_e(0)$.

From a random sample of chromosomes in the population at the time it was considered rescued, average pairwise nucleotide diversity (Tajima's $\pi$), Tajima's $D$, and site-frequency spectra were calculated across 100 adjacent non-overlapping windows (i.e., each of length 200 Kb, i.e., 0.4 cM) using the \texttt{diversity()}, \texttt{Tajimas\char`_D()}, and \texttt{allele\char`_frequency\char`_spectrum()} functions in \texttt{tskit} \cite[][]{kelleher2018efficient}.
We use a sample size of 100 chromosomes throughout.
Linkage disequilibrium was calculated by first identifying the segregating mutation closest to each window's midpoint and then using \texttt{tskit.LdCalculator().r2()} to calculate disequilibrium between it and the segregating mutation closest to a recombination distance of $r=0.001$.
 
For comparison we also run simulations with a constant expected population size, by setting $d=0$.
In this case the ancestral genotype $aa$ has an absolute fitness of 1, meaning that any realized population size trajectory will be a random walk (in our case with an upper boundary at $N(0)$), and that extinction is assured in the long-term.  
However, with the parameter values used here (relatively large initial population size and fast onset of the selective sweeps), population sizes decline only slightly before remaining constant at the carrying capacity once the sweep has started in earnest (since the mutants have fitnesses $\geq 1$).
We chose to use this setup as the constant population size comparison (rather than, say, a Wright-Fisher population) because it allows us to keep the same variance in gamete and offspring numbers (affecting the probability of establishment and the rate of coalescence; see Section \ref{sec:simdetails}) as well as the same census population size (affecting the initial and effective allele frequencies) as in the case of rescue.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Supplementary text: Deriving the structured coalescent}
\label{sec:derive_coalescent}
%
Let the allele frequency and population size $\tau$ generations before the present be $p^\prime(\tau)$ and $N^\prime(\tau)$.
Following \cite{pennings2006soft2}, we artificially subdivide the time within a generation to be able to identify any period between two successive events in our lifecycle (Figure \ref{fig:lifecycle}).
We now go about deriving the probabilities in the structured coalescent (Equation \ref{eq:coal_rates}).
%
\begin{figure}[!htbp]
\centering
\begin{tikzpicture}
%
\def \n {5} %number of stages
\def \nm {4} %number of stages minus 1
\def \radius {3cm} %radius of lifecyle
\def \margin {20} %margin between arrows and circles
\def \names {{"selection", "recombination", "syngamy", "mutation", "migration"}} %names of stages
%
\foreach \s in {0,...,\nm}
{
  \draw[thick] ({360/\n * \s}:\radius) circle[radius=0.75cm]; %circles
  \draw[->, >=latex, thick] ({360/\n * \s + \margin}:\radius) %arrows
    arc ({360/\n * \s + \margin}:{360/\n * (\s+1) - \margin}:\radius);
  \node at  ({360/\n * (\s+0.5)}:\radius+1cm) {\pgfmathparse{\names[\s]}\pgfmathresult}; %names  
  \node at  ({360/\n * \s}:\radius) {\ifthenelse{\s=0}{$\tau$,$\tau-1$}{$\tau -$\s/\n}}; %times
}
\node at (0:\radius+1.5cm) {census};
%
\end{tikzpicture}
\caption{
Life-cycle and time notation.
}%
\label{fig:lifecycle}
\end{figure}
%
\subsection*{Migration}
%
The number of migrant alleles that arrive each generation is Poisson with mean $m$. 
Given that there are $2N^\prime(\tau-1)p^\prime(\tau-1)$ beneficial alleles in the next generation, the probability that any one is a new migrant is therefore $P = m/[2N^\prime(\tau-1)p^\prime(\tau-1)]\approx m/[2N^\prime(\tau)p^\prime(\tau)]$, where the approximation assumes the number of beneficial alleles changes little from one generation to the next.
The probability that at least one of $k$ beneficial alleles is a migrant is $1-(1-P)^k$, which, with rare migration, is approximately 
\begin{equation}
p_\mathrm{mig}(k,\tau) = k \frac{m}{2N^\prime(\tau)p^\prime(\tau)}.
\end{equation}
For a given probability of being replaced by a migrant allele, the rate of migration in a diploid model is half that of the haploid model \citep[equation 15 in][replacing $M$ with $m$]{pennings2006soft2} as there are twice as many resident alleles.
%
\subsection*{Mutation}
%
The number of beneficial alleles after mutation, $2N^\prime(\tau-4/5)p^\prime(\tau-4/5)$, is the number before mutation plus the number of new mutants
\begin{equation}
2N^\prime(\tau-4/5)p^\prime(\tau-4/5) = 2N^\prime(\tau-3/5)p^\prime(\tau-3/5) + u 2N^\prime(\tau-3/5)[1-p^\prime(\tau-3/5)].
\end{equation}
Because the population size does not change during mutation, $N^\prime(\tau-4/5)=N^\prime(\tau-3/5)$, the frequency of beneficial alleles after mutation is simply
\begin{equation}\label{eq:pu}
p^\prime(\tau-4/5) = p^\prime(\tau-3/5) + u [1-p^\prime(\tau-3/5)].
\end{equation}
The probability a beneficial allele is a new mutant is therefore $u [1-p^\prime(\tau-3/5)]/p^\prime(\tau-4/5)$, which, using Equation \ref{eq:pu}, is equivalent to
\begin{equation}
P = \frac{u [1-p^\prime(\tau-4/5)]}{(1-u)p^\prime(\tau-4/5)}.
\end{equation}
The probability that at least one of $k$ beneficial alleles is a new mutant is $1-(1-P)^k$, which, when mutation is rare, is approximately $k u [1-p^\prime(\tau-4/5)]/p^\prime(\tau-4/5)$.
With little change in allele frequency from one generation to the next this is approximately
\begin{equation}
p_\mathrm{mut}(k,\tau) = k u \frac{1-p^\prime(\tau)}{p^\prime(\tau)}.
\end{equation}
This is equivalent to the haploid result \citep[e.g., equation 5 in][]{pennings2006soft2} as both the mutation rate and number of alleles are multiplied by the ploidy level, which cancels.
%
\subsection*{Coalescence}
%
Considering $k$ beneficial alleles at the time of census, and ignoring any migration or mutation, the probability of at least one coalescence event is then 
\begin{equation}
\binom{k}{2}\frac{1}{2N_e'(\tau-2/5)p^\prime(\tau-2/5)},
\end{equation}
where $N_e'(\tau-2/5)$ is the effective population size at the time of syngamy.
When allele frequency and effective population size changes little from one generation to the next this is roughly
\begin{equation}
p_\mathrm{coal}(k,\tau) = \binom{k}{2}\frac{1}{2N_e'(\tau)p^\prime(\tau)}.
\end{equation}
This is half the rate observed in a haploid model with the same population size \citep[equation 5 in][]{pennings2006soft2} as there are twice as many alleles in a diploid population.
%
\subsection*{Recombination}
%
Consider a neutral locus at recombination distance $r$ from the selected site.
Assuming weak selection such that the survivors of viability selection remain in Hardy-Weinberg proportions, the number of alleles linked to the beneficial allele after recombination is
\begin{equation}
\begin{aligned}
2 N^\prime(\tau-2/5) p^\prime(\tau-2/5) &= 2 N^\prime(\tau-1/5) p^\prime(\tau-1/5) \left[p^\prime(\tau-1/5) + \left[1-p^\prime(\tau-1/5) \right] (1-r) \right]\\
 &+  N^\prime(\tau-1/5) [1-p^\prime(\tau-1/5)] p^\prime(\tau-1/5) r.
\end{aligned}
\end{equation}
The first term on the right hand side is the number currently linked to the beneficial allele multiplied by the probability of being in a beneficial homozygote plus the probability of being in a heterozygote but not recombining.
The second term on the right hand side is the number not currently linked with the beneficial allele times the probability of being in a heterozygote and recombining onto the beneficial background.
%
The probability an allele on the beneficial background after recombination was not there before is then
\begin{equation}
P = \frac{2 N^\prime(\tau-1/5) [1-p^\prime(\tau-1/5)] p^\prime(\tau-1/5) r}{2 N^\prime(\tau-2/5) p^\prime(\tau-2/5)},
\end{equation}
which, because recombination does not change allele frequency or population size, is
\begin{equation}
\begin{aligned}
P &= \frac{2 N^\prime(\tau-1/5) [1-p^\prime(\tau-1/5)] p^\prime(\tau-1/5) r}{2 N^\prime(\tau-1/5) p^\prime(\tau-1/5)}\\
&= [1-p^\prime(\tau-1/5)] r.
\end{aligned}
\end{equation}
The probability at least one of $k$ alleles on the beneficial background recombines off is $1-(1-P)^k$, which, when recombination is rare, is approximately $k r [1-p^\prime(\tau-1/5)]$.
Assuming allele frequency changes little through one bout of selection this is $k r [1-p^\prime(\tau)]$.
Finally, assuming migration, mutation, and coalescence are rare, the probability that none of $k$ beneficial alleles migrates or mutates times the probability none coalesce times the probability at least one of the $k$ linked alleles recombines off is roughly \citep[table 1 in][]{hudson1988coalescent}
\begin{equation}
p_\mathrm{rec}(k,\tau) = k r [1-p^\prime(\tau)].
\end{equation}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\newpage
\section*{Supplementary figures}
\setcounter{figure}{0}
\renewcommand{\thefigure}{S\arabic{figure}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}

\node[] (A) {\includegraphics[width=0.49\linewidth]{EH_rescue_s0.2_k100_u0_m0.pdf}};
\node[right = 0 of A] (B) {\includegraphics[width=0.49\linewidth]{EH_rescue_s0.2_k10_u0_m0.pdf}};
\node[below = 0 of A] (C) {\includegraphics[width=0.49\linewidth]{EH_rescue_s0.13_k100_u0_m0.pdf}};
\node[right = 0 of C] (D) {\includegraphics[width=0.49\linewidth]{EH_rescue_s0.13_k10_u0_m0.pdf}};
\node[right = 0 of A, xshift = 0.6cm, yshift = -0.5cm, draw, rectangle, rounded corners, thick, fill=gray!50, rotate = 90] (s) {\large$\bm{s=0.20}$};
\node[right = 0 of C,  xshift = 0.6cm, yshift = -0.5cm, draw, rectangle, rounded corners, thick, fill=gray!50, rotate = 90] {\large$\bm{s=0.13}$};
\node[below = -2/3 of A, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] (k) {\large$\bm{\kappa=100}$};
\node[below = -2/3 of B, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{\kappa=10}$};
\node[right = 2cm of k, darkred, yshift = 0.25cm, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont \textbf{constant $\bm{N}$}};
\node[below = 0 of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont \textbf{evol.\ rescue}};
\node (pt) at ($(D) + (-0.7cm,1.5cm)$) {\footnotesize\fontfamily{phv}\selectfont predicted $\mathbb{E}[\pi | \mathrm{unlinked}]$};
\draw (pt) -- ($(pt)+(0.1,-0.8)$);
\node (nt) at ($(D) + (-0.95cm,-0.5cm)$) {\footnotesize\fontfamily{phv}\selectfont observed $\mathbb{E}[\pi | \mathrm{unlinked}]$};
\draw (nt) -- ($(nt)+(0.1,0.8)$);
\end{tikzpicture}
\caption{
Mean pairwise diversity, $\pi$, after a selective sweep from standing genetic variation in evolutionary rescue (blue; $d=0.05$) or in a population of roughly constant size (red; $d=0$).
The thick solid curves are our approximations (Equation \ref{eq:EH_rescue}).
The dashed blue curves replace $\mathbb{E}[\pi | \mathrm{unlinked}]$ with the observed genome-wide average $\pi$ (excluding sites within 5cM of the selected site as we simulate only a portion of a genome).
The thinnest curves are 100 replicate simulations (rescue only for clarity) and the slightly thicker curves are simulation means (often obscured by prediction).
Parameters: $N(0)=10^4$.
}%
\label{fig:rescueSGV_heterozygosity_empirical}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htbp]
\centering
\begin{tikzpicture}
\node[] (A1) {\includegraphics[width=0.49\linewidth]{pdynamics_rescue_s0.2_k0_u0.0001_m0.pdf}};
\node[below = -0.9cm of A1] (A2) {\includegraphics[width=0.49\linewidth]{ndynamics_rescue_s0.2_k0_u0.0001_m0.pdf}};
\node[right = 0cm of A1] (B1) {\includegraphics[width=0.49\linewidth]{pdynamics_rescue_s0.2_k0_u0.00001_m0.pdf}};
\node[right = 0cm of A2] (B2) {\includegraphics[width=0.49\linewidth]{ndynamics_rescue_s0.2_k0_u0.00001_m0.pdf}};
\node[below = 0cm of A2] (C1) {\includegraphics[width=0.49\linewidth]{pdynamics_rescue_s0.13_k0_u0.0001_m0.pdf}};
\node[below = -0.9cm of C1] (C2) {\includegraphics[width=0.49\linewidth]{ndynamics_rescue_s0.13_k0_u0.0001_m0.pdf}};
\node[right = 0cm of C1] (D1) {\includegraphics[width=0.49\linewidth]{pdynamics_rescue_s0.13_k0_u0.00001_m0.pdf}};
\node[right = 0cm of C2] (D2) {\includegraphics[width=0.49\linewidth]{ndynamics_rescue_s0.13_k0_u0.00001_m0.pdf}};
\node[right = 0.5 of A1, yshift = -2.5cm, rotate = 90, draw, rectangle, rounded corners, thick, fill=gray!50] (s) {\large$\bm{s=0.20}$};
\node[right = 0.5 of C1, yshift = -2.5cm, rotate = 90, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{s=0.13}$};
\node[below = -0.6cm of A2, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] (k) {\large$\bm{u=10^{-4}}$};
\node[below = -0.6cm of B2, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{u=10^{-5}}$};
\node[right = 2cm of k, darkred, yshift = 0.25cm, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont\textbf{constant $\bm{N}$}};
\node[below = 0cm of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont\textbf{evol.\ rescue}};
\end{tikzpicture}
\caption{
Allele frequency, $p^\prime(\tau)$, and population size, $N^\prime(\tau)$, at generation $\tau$ before fixation during a selective sweep from \textit{de novo} mutation in evolutionary rescue (blue; $d=0.05$) and in a population of roughly constant size (red; $d=0$).
The thick solid curves are analytic approximations (Equation \ref{eq:backwardsptnt}), using $p_0 = p_{0|\mathrm{rescue}}^\mathrm{DNM}$ (Equation \ref{eq:p0dnm}) as the initial frequency and $p_f$ (Equation \ref{eq:pf}) as the final allele frequency.
The thin curves are 100 replicate simulations where a sweep (and population recovery) was observed.
The stars show the predicted time to fixation (Equation \ref{eq:tfix}).
The arrows show the mean time to fixation observed in simulations. 
}%
\label{fig:rescueDNM_dynamics}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}
\node[] (A1) {\includegraphics[width=0.49\linewidth]{coalescent_rescue_s0.2_k0_u0.0001_m0.pdf}};
\node[right = 0cm of A1] (B1) {\includegraphics[width=0.49\linewidth]{coalescent_rescue_s0.2_k0_u0.00001_m0.pdf}};
\node[below = 0cm of A1] (C1) {\includegraphics[width=0.49\linewidth]{coalescent_rescue_s0.13_k0_u0.0001_m0.pdf}};
\node[right = 0cm of C1] (D1) {\includegraphics[width=0.49\linewidth]{coalescent_rescue_s0.13_k0_u0.00001_m0.pdf}};
\node[right = 0.5 of A1, yshift = -0.5cm, rotate = 90, draw, rectangle, rounded corners, thick, fill=gray!50] (s) {\large$\bm{s=0.20}$};
\node[right = 0.5 of C1, yshift = -0.5cm, rotate = 90, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{s=0.13}$};
\node[below = -0.6cm of A1, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] (k) {\large$\bm{u=10^{-4}}$};
\node[below = -0.6cm of B1, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{u=10^{-5}}$};
\node[right = 2cm of k, darkred, yshift = 0.25cm, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont\textbf{constant $\bm{N}$}};
\node[below = 0cm of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont\textbf{evol.\ rescue}};
\node (pt) at ($(A1) + (3cm,1cm)$) {\footnotesize\fontfamily{phv}\selectfont coalescence};
\draw (pt) -- ($(pt)+(-1,-2)$);
\node (nt) at ($(A1) + (0cm,1cm)$) {\footnotesize\fontfamily{phv}\selectfont recombination};
\draw (nt) -- ($(nt)+(1.5,-1/2)$);
\node (mut) at ($(A1) + (-1cm,0cm)$) {\footnotesize\fontfamily{phv}\selectfont mutation};
\draw (mut) -- ($(mut)+(1.8,-0.1)$);
\end{tikzpicture}
\caption{
The timing of events in the structured coalescent (Equation \ref{eq:Pikt}) for a sample of size $k=2$ at a linked neutral locus ($r=0.01$) during a selective sweep from  \textit{de novo} mutation in evolutionary rescue (blue; $d=0.05$) or in a population of roughly constant size (red; $d=0$).
Thicker opaque curves use the analytic expressions for allele frequency and population size (Equation \ref{eq:backwardsptnt}) while the thinner transparent curves show the mean probabilities given the observed allele frequency and population size dynamics in 100 replicate simulations (these become more variable with time as less replicates remain polymorphic).
}%
\label{fig:rescueDNM_timing}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}
\node[] (A) {\includegraphics[width=0.49\linewidth]{EH_rescue_s0.2_k0_u0.0001_m0_relative.pdf}};
\node[right = 0 of A] (B) {\includegraphics[width=0.49\linewidth]{EH_rescue_s0.2_k0_u0.00001_m0_relative.pdf}};
\node[below = 0 of A] (C) {\includegraphics[width=0.49\linewidth]{EH_rescue_s0.13_k0_u0.0001_m0_relative.pdf}};
\node[right = 0 of C] (D) {\includegraphics[width=0.49\linewidth]{EH_rescue_s0.13_k0_u0.00001_m0_relative.pdf}};
\node[right = 0 of A, xshift = 0.6cm, yshift = -0.5cm, draw, rectangle, rounded corners, thick, fill=gray!50, rotate = 90] (s) {\large$\bm{s=0.20}$};
\node[right = 0 of C,  xshift = 0.6cm, yshift = -0.5cm, draw, rectangle, rounded corners, thick, fill=gray!50, rotate = 90] {\large$\bm{s=0.13}$};
\node[below = -2/3 of A, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] (k) {\large$\bm{u=10^{-4}}$};
\node[below = -2/3 of B, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{u=10^{-5}}$};
\node[right = 2cm of k, darkred, yshift = 0.25cm, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont \textbf{constant $\bm{N}$}};
\node[below = 0 of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont \textbf{evol.\ rescue}};
\end{tikzpicture}
\caption{
Relative mean pairwise diversity, $\pi/\overline{\pi}$, after a selective sweep from \textit{de novo} mutation during evolutionary rescue (blue; $d=0.05$) or in a population of roughly constant size (red; $d=0$).
The thickest curves are $\mathbb{E}[\pi]/\mathbb{E}[\pi|\mathrm{unlinked}]$ (using Equation \ref{eq:EH_rescue}).
The thinnest curves are 100 replicate simulations (rescue only for clarity) and the slightly thicker curves are simulation means (often obscured by prediction).
Parameters: $N(0)=10^4$.
}%
\label{fig:rescueDNM_heterozygosity_relative}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}
\node[] (A) {\includegraphics[width=0.49\linewidth]{TajimasD_rescue_s0.2_k0_u0.0001_m0.pdf}};
\node[right =0 of A] (B) {\includegraphics[width=0.49\linewidth]{TajimasD_rescue_s0.2_k0_u0.00001_m0.pdf}};
\node[below =0 of A] (C) {\includegraphics[width=0.49\linewidth]{TajimasD_rescue_s0.13_k0_u0.0001_m0.pdf}};
\node[right =0 of C] (D) {\includegraphics[width=0.49\linewidth]{TajimasD_rescue_s0.13_k0_u0.00001_m0.pdf}};
\node[right = 0 of A, xshift = 0.6cm, yshift = -0.5cm, draw, rectangle, rounded corners, thick, fill=gray!50, rotate = 90] (s) {\large$\bm{s=0.20}$};
\node[right = 0 of C,  xshift = 0.6cm, yshift = -0.5cm, draw, rectangle, rounded corners, thick, fill=gray!50, rotate = 90] {\large$\bm{s=0.13}$};
\node[below = -2/3 of A, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] (k) {\large$\bm{u=10^{-4}}$};
\node[below = -2/3 of B, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{u=10^{-5}}$};
\node[right = 2cm of k, darkred, yshift = 0.25cm, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont \textbf{constant $\bm{N}$}};
\node[below = 0 of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont \textbf{evol.\ rescue}};
\end{tikzpicture}
\caption{
Tajima's $D$ after a selective sweep from \textit{de novo} mutation during evolutionary rescue (blue; $d=0.05$) or in a population of roughly constant size (red; $d=0$).
Thin curves show 100 replicate simulations (rescue only for clarity) and thicker curves show simulation means.
Parameters: $N(0)=10^4$.
}%
\label{fig:rescueDNM_tajimasD}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htbp]
\centering
\begin{tikzpicture}
\node[] (A1) {\includegraphics[width=0.49\linewidth]{pdynamics_rescue_s0.2_k0_u0_m1.pdf}};
\node[below = -0.9cm of A1] (A2) {\includegraphics[width=0.49\linewidth]{ndynamics_rescue_s0.2_k0_u0_m1.pdf}};
\node[right = 0cm of A1] (B1) {\includegraphics[width=0.49\linewidth]{pdynamics_rescue_s0.2_k0_u0_m0.1.pdf}};
\node[right = 0cm of A2] (B2) {\includegraphics[width=0.49\linewidth]{ndynamics_rescue_s0.2_k0_u0_m0.1.pdf}};
\node[below = 0cm of A2] (C1) {\includegraphics[width=0.49\linewidth]{pdynamics_rescue_s0.13_k0_u0_m1.pdf}};
\node[below = -0.9cm of C1] (C2) {\includegraphics[width=0.49\linewidth]{ndynamics_rescue_s0.13_k0_u0_m1.pdf}};
\node[right = 0cm of C1] (D1) {\includegraphics[width=0.49\linewidth]{pdynamics_rescue_s0.13_k0_u0_m0.1.pdf}};
\node[right = 0cm of C2] (D2) {\includegraphics[width=0.49\linewidth]{ndynamics_rescue_s0.13_k0_u0_m0.1.pdf}};
\node[right = 0.5 of A1, yshift = -2.5cm, rotate = 90, draw, rectangle, rounded corners, thick, fill=gray!50] (s) {\large$\bm{s=0.20}$};
\node[right = 0.5 of C1, yshift = -2.5cm, rotate = 90, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{s=0.13}$};
\node[below = -0.6cm of A2, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] (k) {\large$\bm{m=1}$};
\node[below = -0.6cm of B2, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{m=0.1}$};
\node[right = 2cm of k, darkred, yshift = 0.25cm, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont\textbf{constant $\bm{N}$}};
\node[below = 0cm of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont\textbf{evol.\ rescue}};
\end{tikzpicture}
\caption{
Allele frequency and population size during a selective sweep from migration in evolutionary rescue (blue; $d=0.05$) or in a population of roughly constant size (red; $d=0$).
See Figure \ref{fig:rescueSGV_dynamics} for details.
}%
\label{fig:rescueMIG_dynamics}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{figure}[htb]
\centering
\begin{tikzpicture}
\node[] (A1) {\includegraphics[width=0.49\linewidth]{coalescent_rescue_s0.2_k0_u0_m1.pdf}};
\node[right = 0cm of A1] (B1) {\includegraphics[width=0.49\linewidth]{coalescent_rescue_s0.2_k0_u0_m0.1.pdf}};
\node[below = 0cm of A1] (C1) {\includegraphics[width=0.49\linewidth]{coalescent_rescue_s0.13_k0_u0_m1.pdf}};
\node[right = 0cm of C1] (D1) {\includegraphics[width=0.49\linewidth]{coalescent_rescue_s0.13_k0_u0_m0.1.pdf}};
\node[right = 0.5 of A1, yshift = -0.5cm, rotate = 90, draw, rectangle, rounded corners, thick, fill=gray!50] (s) {\large$\bm{s=0.20}$};
\node[right = 0.5 of C1, yshift = -0.5cm, rotate = 90, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{s=0.13}$};
\node[below = -0.6cm of A1, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] (k) {\large$\bm{m=1}$};
\node[below = -0.6cm of B1, xshift = 20, draw, rectangle, rounded corners, thick, fill=gray!50] {\large$\bm{m=1}$};
\node[right = 2cm of k, darkred, yshift = 0.25cm, draw, rectangle, rounded corners, thick, fill=darkred!25] (N) {\fontfamily{phv}\selectfont\textbf{constant $\bm{N}$}};
\node[below = 0cm of N, darkblue, draw, rectangle, rounded corners, thick, fill=darkblue!25] {\fontfamily{phv}\selectfont\textbf{evol.\ rescue}};
\node (pt) at ($(B1) + (3cm,1cm)$) {\footnotesize\fontfamily{phv}\selectfont coalescence};
\draw (pt) -- ($(pt)+(-1,-1.8)$);
\node (nt) at ($(B1) + (0cm,1cm)$) {\footnotesize\fontfamily{phv}\selectfont recombination};
\draw (nt) -- ($(nt)+(1.5,-1/2)$);
\node (mut) at ($(B1) + (-1cm,-1cm)$) {\footnotesize\fontfamily{phv}\selectfont migration};
\draw (mut) -- ($(mut)+(1.5,-0.6)$);
\end{tikzpicture}
\caption{
The timing of events in the structured coalescent (Equation \ref{eq:Pikt}) for a sample of size $k=2$ at a linked neutral locus ($r=0.01$) during a selective sweep from  migration in evolutionary rescue (blue; $d=0.05$) or in a population of roughly constant size (red; $d=0$).
Thicker opaque curves use the analytic expressions for allele frequency and population size (Equation \ref{eq:backwardsptnt}) while the thinner transparent curves show the mean probabilities given the observed allele frequency and population size dynamics in 100 replicate simulations (these become more variable with time as less replicates remain polymorphic).
}%
\label{fig:rescueMIG_timing}
\end{figure}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\end{document}